{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import gseapy\n",
    "import math\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as ticker\n",
    "import numpy as np\n",
    "import os\n",
    "import pandas as pd\n",
    "import re\n",
    "import scipy.stats as stats\n",
    "import seaborn as sns\n",
    "import sys\n",
    "import time\n",
    "\n",
    "from scipy.stats import spearmanr\n",
    "from scipy.stats import pearsonr\n",
    "from statsmodels.sandbox.stats import multicomp\n",
    "\n",
    "# import utils\n",
    "sys.path.append(\"../../../utils\")\n",
    "from plotting_utils import *\n",
    "from classify_utils import *\n",
    "\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'svg'\n",
    "mpl.rcParams['figure.autolayout'] = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.set(**PAPER_PRESET)\n",
    "fontsize = PAPER_FONTSIZE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "expr_dir = \"../../../misc/07__ENCODE_expr\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_sub = \"w_gene\"\n",
    "tx_sub = \"tx_only\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# seq_counts_f = \"../../../data/00__rna_seq/00__sleuth_results/sleuth_abundances_raw_counts.TRANSCRIPTS_WITH_GENE_ISOFORM.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_map_f = \"../../../misc/00__gene_metadata/gencode.v25lift37.GENE_ID_TRANSCRIPT_ID_MAP.with_DIGIT.fixed.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "rna_seq_f = \"../../../data/00__rna_seq/01__processed_results/rna_seq_results.tsv\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'A549_1_1',\n",
       " 'A549_2_1',\n",
       " 'A549_3_1',\n",
       " 'A549_4_1',\n",
       " 'GM12878_1_1',\n",
       " 'GM12878_2_1',\n",
       " 'H1_1_1',\n",
       " 'HT1080_1_1',\n",
       " 'HT1080_2_1',\n",
       " 'HUES64_1_3',\n",
       " 'IMR-90_1_1',\n",
       " 'IMR-90_2_1',\n",
       " 'K562_1_1',\n",
       " 'K562_2_1',\n",
       " 'MCF-7_3_1',\n",
       " 'MCF-7_4_1',\n",
       " 'NCI-H460_1_1',\n",
       " 'NCI-H460_2_1',\n",
       " 'SK-MEL-5_1_1',\n",
       " 'SK-MEL-5_2_1',\n",
       " 'SK-N-DZ_1_1',\n",
       " 'SK-N-DZ_2_1',\n",
       " 'SK-N-SH_3_1',\n",
       " 'SK-N-SH_4_1'}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = [x[2] for x in os.walk(expr_dir)]\n",
    "samps = set([x.split(\".\")[0] for x in files[0] if \"kallisto\" in x])\n",
    "samps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A549_4_1\n",
      "K562_1_1\n",
      "H1_1_1\n",
      "NCI-H460_2_1\n",
      "SK-N-DZ_1_1\n",
      "SK-MEL-5_1_1\n",
      "NCI-H460_1_1\n",
      "SK-N-SH_4_1\n",
      "HT1080_2_1\n",
      "IMR-90_2_1\n",
      "MCF-7_4_1\n",
      "MCF-7_3_1\n",
      "HUES64_1_3\n",
      "A549_3_1\n",
      "GM12878_2_1\n",
      "SK-N-DZ_2_1\n",
      "SK-N-SH_3_1\n",
      "A549_1_1\n",
      "HT1080_1_1\n",
      "A549_2_1\n",
      "K562_2_1\n",
      "GM12878_1_1\n",
      "IMR-90_1_1\n",
      "SK-MEL-5_2_1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target_id</th>\n",
       "      <th>A549_4_1</th>\n",
       "      <th>K562_1_1</th>\n",
       "      <th>H1_1_1</th>\n",
       "      <th>NCI-H460_2_1</th>\n",
       "      <th>SK-N-DZ_1_1</th>\n",
       "      <th>SK-MEL-5_1_1</th>\n",
       "      <th>NCI-H460_1_1</th>\n",
       "      <th>SK-N-SH_4_1</th>\n",
       "      <th>HT1080_2_1</th>\n",
       "      <th>...</th>\n",
       "      <th>GM12878_2_1</th>\n",
       "      <th>SK-N-DZ_2_1</th>\n",
       "      <th>SK-N-SH_3_1</th>\n",
       "      <th>A549_1_1</th>\n",
       "      <th>HT1080_1_1</th>\n",
       "      <th>A549_2_1</th>\n",
       "      <th>K562_2_1</th>\n",
       "      <th>GM12878_1_1</th>\n",
       "      <th>IMR-90_1_1</th>\n",
       "      <th>SK-MEL-5_2_1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000456328.2_1</td>\n",
       "      <td>0.763808</td>\n",
       "      <td>0.313936</td>\n",
       "      <td>0.004047</td>\n",
       "      <td>1.235340</td>\n",
       "      <td>4.806000</td>\n",
       "      <td>1.15116</td>\n",
       "      <td>0.659146</td>\n",
       "      <td>0.734967</td>\n",
       "      <td>0.691339</td>\n",
       "      <td>...</td>\n",
       "      <td>0.034047</td>\n",
       "      <td>5.64779</td>\n",
       "      <td>0.334105</td>\n",
       "      <td>0.573798</td>\n",
       "      <td>0.487667</td>\n",
       "      <td>0.156877</td>\n",
       "      <td>0.325951</td>\n",
       "      <td>0.017293</td>\n",
       "      <td>0.072287</td>\n",
       "      <td>0.769937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000450305.2_1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.019143</td>\n",
       "      <td>0.341847</td>\n",
       "      <td>0.046717</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.031573</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002571</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.048903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000488147.1_1</td>\n",
       "      <td>11.502300</td>\n",
       "      <td>13.862000</td>\n",
       "      <td>4.382520</td>\n",
       "      <td>38.673200</td>\n",
       "      <td>68.040800</td>\n",
       "      <td>40.79530</td>\n",
       "      <td>44.566400</td>\n",
       "      <td>19.772600</td>\n",
       "      <td>23.152300</td>\n",
       "      <td>...</td>\n",
       "      <td>3.035160</td>\n",
       "      <td>50.83590</td>\n",
       "      <td>18.962000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>22.800500</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>13.735500</td>\n",
       "      <td>2.863410</td>\n",
       "      <td>15.670100</td>\n",
       "      <td>63.416700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.028021</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.045354</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.062664</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000469289.1_1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.033696</td>\n",
       "      <td>0.110995</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.035944</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.009972</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.033052</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             target_id   A549_4_1   K562_1_1    H1_1_1  NCI-H460_2_1  \\\n",
       "0  ENST00000456328.2_1   0.763808   0.313936  0.004047      1.235340   \n",
       "1  ENST00000450305.2_1   0.000000   0.000000  0.019143      0.341847   \n",
       "2  ENST00000488147.1_1  11.502300  13.862000  4.382520     38.673200   \n",
       "3  ENST00000473358.1_1   0.000000   0.000000  0.028021      0.000000   \n",
       "4  ENST00000469289.1_1   0.000000   0.033696  0.110995      0.000000   \n",
       "\n",
       "   SK-N-DZ_1_1  SK-MEL-5_1_1  NCI-H460_1_1  SK-N-SH_4_1  HT1080_2_1  \\\n",
       "0     4.806000       1.15116      0.659146     0.734967    0.691339   \n",
       "1     0.046717       0.00000      0.000000     0.000000    0.000000   \n",
       "2    68.040800      40.79530     44.566400    19.772600   23.152300   \n",
       "3     0.000000       0.00000      0.000000     0.000000    0.000000   \n",
       "4     0.000000       0.00000      0.000000     0.035944    0.000000   \n",
       "\n",
       "       ...       GM12878_2_1  SK-N-DZ_2_1  SK-N-SH_3_1  A549_1_1  HT1080_1_1  \\\n",
       "0      ...          0.034047      5.64779     0.334105  0.573798    0.487667   \n",
       "1      ...          0.031573      0.00000     0.000000  0.000000    0.000000   \n",
       "2      ...          3.035160     50.83590    18.962000  0.000000   22.800500   \n",
       "3      ...          0.000000      0.00000     0.045354  0.000000    0.000000   \n",
       "4      ...          0.009972      0.00000     0.000000  0.000000    0.000000   \n",
       "\n",
       "   A549_2_1   K562_2_1  GM12878_1_1  IMR-90_1_1  SK-MEL-5_2_1  \n",
       "0  0.156877   0.325951     0.017293    0.072287      0.769937  \n",
       "1  0.000000   0.000000     0.002571    0.000000      0.048903  \n",
       "2  0.000000  13.735500     2.863410   15.670100     63.416700  \n",
       "3  0.000000   0.062664     0.000000    0.000000      0.000000  \n",
       "4  0.000000   0.033052     0.000000    0.000000      0.000000  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for i, samp in enumerate(samps):\n",
    "    print(samp)\n",
    "    tx_path = \"%s/%s.%s.kallisto.tsv\" % (expr_dir, samp, tx_sub)\n",
    "    gene_path = \"%s/%s.%s.kallisto.tsv\" % (expr_dir, samp, gene_sub)\n",
    "    if i == 0:\n",
    "        tx_df = pd.read_table(tx_path, sep=\"\\t\")[[\"target_id\", \"tpm\"]]\n",
    "        tx_df.columns = [\"target_id\", samp]\n",
    "        gene_df = pd.read_table(gene_path, sep=\"\\t\")[[\"target_id\", \"tpm\"]]\n",
    "        gene_df.columns = [\"target_id\", samp]\n",
    "    else:\n",
    "        tx_tmp = pd.read_table(tx_path, sep=\"\\t\")[[\"target_id\", \"tpm\"]]\n",
    "        tx_tmp.columns = [\"target_id\", samp]\n",
    "        gene_tmp = pd.read_table(gene_path, sep=\"\\t\")[[\"target_id\", \"tpm\"]]\n",
    "        gene_tmp.columns = [\"target_id\", samp]\n",
    "        \n",
    "        tx_df = tx_df.merge(tx_tmp, on=\"target_id\")\n",
    "        gene_df = gene_df.merge(gene_tmp, on=\"target_id\")\n",
    "\n",
    "tx_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# seq_counts = pd.read_table(seq_counts_f).reset_index()\n",
    "# seq_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972.5_1</td>\n",
       "      <td>ENST00000456328.2_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000223972.5_1</td>\n",
       "      <td>ENST00000450305.2_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000227232.5_1</td>\n",
       "      <td>ENST00000488147.1_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000243485.4_2</td>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000243485.4_2</td>\n",
       "      <td>ENST00000469289.1_1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id        transcript_id\n",
       "0  ENSG00000223972.5_1  ENST00000456328.2_1\n",
       "1  ENSG00000223972.5_1  ENST00000450305.2_1\n",
       "2  ENSG00000227232.5_1  ENST00000488147.1_1\n",
       "3  ENSG00000243485.4_2  ENST00000473358.1_1\n",
       "4  ENSG00000243485.4_2  ENST00000469289.1_1"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_map = pd.read_table(gene_map_f, header=None)\n",
    "gene_map.columns = [\"gene_id\", \"transcript_id\"]\n",
    "gene_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "170623\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>csf</th>\n",
       "      <th>cleaner_gene_biotype</th>\n",
       "      <th>cleaner_transcript_biotype</th>\n",
       "      <th>hESC_rep1</th>\n",
       "      <th>hESC_rep2</th>\n",
       "      <th>endo_rep1</th>\n",
       "      <th>endo_rep2</th>\n",
       "      <th>...</th>\n",
       "      <th>meso_rep2</th>\n",
       "      <th>hESC_mean</th>\n",
       "      <th>endo_mean</th>\n",
       "      <th>meso_mean</th>\n",
       "      <th>overall_mean</th>\n",
       "      <th>threshold</th>\n",
       "      <th>qval_hESC_endo</th>\n",
       "      <th>qval_hESC_meso</th>\n",
       "      <th>endo_hESC_log2fc</th>\n",
       "      <th>meso_hESC_log2fc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DIGIT</td>\n",
       "      <td>DIGIT</td>\n",
       "      <td>DIGIT</td>\n",
       "      <td>lncRNA_good_csf</td>\n",
       "      <td>promoter_overlap</td>\n",
       "      <td>promoter_overlap</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.025422</td>\n",
       "      <td>4.907468</td>\n",
       "      <td>4.784695</td>\n",
       "      <td>...</td>\n",
       "      <td>0.014741</td>\n",
       "      <td>0.012711</td>\n",
       "      <td>4.846082</td>\n",
       "      <td>0.007370</td>\n",
       "      <td>1.622054</td>\n",
       "      <td>expressed</td>\n",
       "      <td>0.008168525422325441</td>\n",
       "      <td>sleuth NA</td>\n",
       "      <td>2.529248</td>\n",
       "      <td>-0.007628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000000233.9_1</td>\n",
       "      <td>ENSG00000004059.10_2</td>\n",
       "      <td>ARF5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>101.476958</td>\n",
       "      <td>102.386195</td>\n",
       "      <td>222.677573</td>\n",
       "      <td>197.586091</td>\n",
       "      <td>...</td>\n",
       "      <td>245.318970</td>\n",
       "      <td>101.931577</td>\n",
       "      <td>210.131832</td>\n",
       "      <td>204.374142</td>\n",
       "      <td>172.145850</td>\n",
       "      <td>expressed</td>\n",
       "      <td>0.0057874417654439295</td>\n",
       "      <td>0.0346694249326972</td>\n",
       "      <td>1.036458</td>\n",
       "      <td>0.996569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000000412.3</td>\n",
       "      <td>ENSG00000003056.3</td>\n",
       "      <td>M6PR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>49.696030</td>\n",
       "      <td>46.457026</td>\n",
       "      <td>31.608496</td>\n",
       "      <td>36.350968</td>\n",
       "      <td>...</td>\n",
       "      <td>10.951882</td>\n",
       "      <td>48.076528</td>\n",
       "      <td>33.979732</td>\n",
       "      <td>20.552771</td>\n",
       "      <td>34.203010</td>\n",
       "      <td>expressed</td>\n",
       "      <td>0.00549724038022104</td>\n",
       "      <td>0.13894723887027402</td>\n",
       "      <td>-0.488514</td>\n",
       "      <td>-1.187160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000000442.10_1</td>\n",
       "      <td>ENSG00000173153.13_2</td>\n",
       "      <td>ESRRA</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1.924618</td>\n",
       "      <td>3.493033</td>\n",
       "      <td>3.499700</td>\n",
       "      <td>2.419880</td>\n",
       "      <td>...</td>\n",
       "      <td>7.769760</td>\n",
       "      <td>2.708826</td>\n",
       "      <td>2.959790</td>\n",
       "      <td>7.600094</td>\n",
       "      <td>4.422903</td>\n",
       "      <td>expressed</td>\n",
       "      <td>0.814911561863725</td>\n",
       "      <td>0.0241642757995681</td>\n",
       "      <td>0.094461</td>\n",
       "      <td>1.213390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000001008.5_1</td>\n",
       "      <td>ENSG00000004478.7_1</td>\n",
       "      <td>FKBP4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>71.840722</td>\n",
       "      <td>79.489420</td>\n",
       "      <td>33.770398</td>\n",
       "      <td>33.810736</td>\n",
       "      <td>...</td>\n",
       "      <td>68.012268</td>\n",
       "      <td>75.665071</td>\n",
       "      <td>33.790567</td>\n",
       "      <td>61.985019</td>\n",
       "      <td>57.146886</td>\n",
       "      <td>expressed</td>\n",
       "      <td>0.00160853874236564</td>\n",
       "      <td>0.0935556001786738</td>\n",
       "      <td>-1.139873</td>\n",
       "      <td>-0.283561</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          transcript_id               gene_id gene_name              csf  \\\n",
       "0                 DIGIT                 DIGIT     DIGIT  lncRNA_good_csf   \n",
       "1   ENST00000000233.9_1  ENSG00000004059.10_2      ARF5   protein_coding   \n",
       "2     ENST00000000412.3     ENSG00000003056.3      M6PR   protein_coding   \n",
       "3  ENST00000000442.10_1  ENSG00000173153.13_2     ESRRA   protein_coding   \n",
       "4   ENST00000001008.5_1   ENSG00000004478.7_1     FKBP4   protein_coding   \n",
       "\n",
       "  cleaner_gene_biotype cleaner_transcript_biotype   hESC_rep1   hESC_rep2  \\\n",
       "0     promoter_overlap           promoter_overlap    0.000000    0.025422   \n",
       "1       protein_coding             protein_coding  101.476958  102.386195   \n",
       "2       protein_coding             protein_coding   49.696030   46.457026   \n",
       "3       protein_coding             protein_coding    1.924618    3.493033   \n",
       "4       protein_coding             protein_coding   71.840722   79.489420   \n",
       "\n",
       "    endo_rep1   endo_rep2        ...          meso_rep2   hESC_mean  \\\n",
       "0    4.907468    4.784695        ...           0.014741    0.012711   \n",
       "1  222.677573  197.586091        ...         245.318970  101.931577   \n",
       "2   31.608496   36.350968        ...          10.951882   48.076528   \n",
       "3    3.499700    2.419880        ...           7.769760    2.708826   \n",
       "4   33.770398   33.810736        ...          68.012268   75.665071   \n",
       "\n",
       "    endo_mean   meso_mean  overall_mean  threshold         qval_hESC_endo  \\\n",
       "0    4.846082    0.007370      1.622054  expressed   0.008168525422325441   \n",
       "1  210.131832  204.374142    172.145850  expressed  0.0057874417654439295   \n",
       "2   33.979732   20.552771     34.203010  expressed    0.00549724038022104   \n",
       "3    2.959790    7.600094      4.422903  expressed      0.814911561863725   \n",
       "4   33.790567   61.985019     57.146886  expressed    0.00160853874236564   \n",
       "\n",
       "        qval_hESC_meso endo_hESC_log2fc  meso_hESC_log2fc  \n",
       "0            sleuth NA         2.529248         -0.007628  \n",
       "1   0.0346694249326972         1.036458          0.996569  \n",
       "2  0.13894723887027402        -0.488514         -1.187160  \n",
       "3   0.0241642757995681         0.094461          1.213390  \n",
       "4   0.0935556001786738        -1.139873         -0.283561  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna_seq = pd.read_table(rna_seq_f)\n",
    "print(len(rna_seq))\n",
    "rna_seq.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. sep gene & transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "60203\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target_id</th>\n",
       "      <th>A549_4_1</th>\n",
       "      <th>K562_1_1</th>\n",
       "      <th>H1_1_1</th>\n",
       "      <th>NCI-H460_2_1</th>\n",
       "      <th>SK-N-DZ_1_1</th>\n",
       "      <th>SK-MEL-5_1_1</th>\n",
       "      <th>NCI-H460_1_1</th>\n",
       "      <th>SK-N-SH_4_1</th>\n",
       "      <th>HT1080_2_1</th>\n",
       "      <th>...</th>\n",
       "      <th>GM12878_2_1</th>\n",
       "      <th>SK-N-DZ_2_1</th>\n",
       "      <th>SK-N-SH_3_1</th>\n",
       "      <th>A549_1_1</th>\n",
       "      <th>HT1080_1_1</th>\n",
       "      <th>A549_2_1</th>\n",
       "      <th>K562_2_1</th>\n",
       "      <th>GM12878_1_1</th>\n",
       "      <th>IMR-90_1_1</th>\n",
       "      <th>SK-MEL-5_2_1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>250042</th>\n",
       "      <td>ENSG00000230066.5_1::chrY:9334896-9344176(-)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001442</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232139</th>\n",
       "      <td>ENSG00000083544.13_2::chr13:60970590-61148012(+)</td>\n",
       "      <td>0.026318</td>\n",
       "      <td>0.028148</td>\n",
       "      <td>0.025221</td>\n",
       "      <td>0.239706</td>\n",
       "      <td>0.115286</td>\n",
       "      <td>0.416561</td>\n",
       "      <td>0.149044</td>\n",
       "      <td>0.07698</td>\n",
       "      <td>0.146016</td>\n",
       "      <td>...</td>\n",
       "      <td>0.05719</td>\n",
       "      <td>0.118468</td>\n",
       "      <td>0.036314</td>\n",
       "      <td>0.033869</td>\n",
       "      <td>0.132155</td>\n",
       "      <td>0.02686</td>\n",
       "      <td>0.023689</td>\n",
       "      <td>0.054922</td>\n",
       "      <td>0.012707</td>\n",
       "      <td>0.414472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212835</th>\n",
       "      <td>ENSG00000223597.2_1::chr5:7373229-7379903(+)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246372</th>\n",
       "      <td>ENSG00000182591.5_1::chr21:32252963-32253875(-)</td>\n",
       "      <td>0.020310</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004515</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.032669</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012957</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004574</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247050</th>\n",
       "      <td>ENSG00000253935.1_1::chr22:22574038-22574525(+)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               target_id  A549_4_1  K562_1_1  \\\n",
       "250042      ENSG00000230066.5_1::chrY:9334896-9344176(-)  0.000000  0.000000   \n",
       "232139  ENSG00000083544.13_2::chr13:60970590-61148012(+)  0.026318  0.028148   \n",
       "212835      ENSG00000223597.2_1::chr5:7373229-7379903(+)  0.000000  0.000000   \n",
       "246372   ENSG00000182591.5_1::chr21:32252963-32253875(-)  0.020310  0.000000   \n",
       "247050   ENSG00000253935.1_1::chr22:22574038-22574525(+)  0.000000  0.000000   \n",
       "\n",
       "          H1_1_1  NCI-H460_2_1  SK-N-DZ_1_1  SK-MEL-5_1_1  NCI-H460_1_1  \\\n",
       "250042  0.000000      0.000000     0.000000      0.000000      0.000000   \n",
       "232139  0.025221      0.239706     0.115286      0.416561      0.149044   \n",
       "212835  0.000000      0.000000     0.000000      0.000000      0.000000   \n",
       "246372  0.004515      0.000000     0.000000      0.032669      0.000000   \n",
       "247050  0.000000      0.000000     0.000000      0.000000      0.000000   \n",
       "\n",
       "        SK-N-SH_4_1  HT1080_2_1      ...       GM12878_2_1  SK-N-DZ_2_1  \\\n",
       "250042      0.00000    0.000000      ...           0.00000     0.000000   \n",
       "232139      0.07698    0.146016      ...           0.05719     0.118468   \n",
       "212835      0.00000    0.000000      ...           0.00000     0.000000   \n",
       "246372      0.00000    0.000000      ...           0.00000     0.000000   \n",
       "247050      0.00000    0.000000      ...           0.00000     0.000000   \n",
       "\n",
       "        SK-N-SH_3_1  A549_1_1  HT1080_1_1  A549_2_1  K562_2_1  GM12878_1_1  \\\n",
       "250042     0.000000  0.000000    0.000000   0.00000  0.000000     0.000000   \n",
       "232139     0.036314  0.033869    0.132155   0.02686  0.023689     0.054922   \n",
       "212835     0.000000  0.000000    0.000000   0.00000  0.000000     0.000000   \n",
       "246372     0.012957  0.000000    0.000000   0.00000  0.000000     0.000000   \n",
       "247050     0.000000  0.000000    0.000000   0.00000  0.000000     0.000000   \n",
       "\n",
       "        IMR-90_1_1  SK-MEL-5_2_1  \n",
       "250042    0.001442      0.000000  \n",
       "232139    0.012707      0.414472  \n",
       "212835    0.000000      0.000000  \n",
       "246372    0.004574      0.000000  \n",
       "247050    0.000000      0.000000  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_counts = gene_df[gene_df[\"target_id\"].str.startswith(\"ENSG\")]\n",
    "print(len(gene_counts))\n",
    "gene_counts.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "60204\n"
     ]
    }
   ],
   "source": [
    "gene_counts = gene_counts.append(gene_df[gene_df[\"target_id\"].str.contains(\"DIGIT::\")])\n",
    "print(len(gene_counts))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "200139\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target_id</th>\n",
       "      <th>A549_4_1</th>\n",
       "      <th>K562_1_1</th>\n",
       "      <th>H1_1_1</th>\n",
       "      <th>NCI-H460_2_1</th>\n",
       "      <th>SK-N-DZ_1_1</th>\n",
       "      <th>SK-MEL-5_1_1</th>\n",
       "      <th>NCI-H460_1_1</th>\n",
       "      <th>SK-N-SH_4_1</th>\n",
       "      <th>HT1080_2_1</th>\n",
       "      <th>...</th>\n",
       "      <th>GM12878_2_1</th>\n",
       "      <th>SK-N-DZ_2_1</th>\n",
       "      <th>SK-N-SH_3_1</th>\n",
       "      <th>A549_1_1</th>\n",
       "      <th>HT1080_1_1</th>\n",
       "      <th>A549_2_1</th>\n",
       "      <th>K562_2_1</th>\n",
       "      <th>GM12878_1_1</th>\n",
       "      <th>IMR-90_1_1</th>\n",
       "      <th>SK-MEL-5_2_1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>97645</th>\n",
       "      <td>ENST00000502161.6_1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.022999</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.073156</td>\n",
       "      <td>0.071750</td>\n",
       "      <td>0.048549</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007496</td>\n",
       "      <td>0.014783</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165675</th>\n",
       "      <td>ENST00000528059.1_1</td>\n",
       "      <td>0.032773</td>\n",
       "      <td>0.975492</td>\n",
       "      <td>0.362165</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.319043</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.029940</td>\n",
       "      <td>0.024811</td>\n",
       "      <td>...</td>\n",
       "      <td>0.031460</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.083639</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.325053</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.819158</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.008877</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13838</th>\n",
       "      <td>ENST00000255416.8_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.015455</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.018436</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.043335</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.029581</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44299</th>\n",
       "      <td>ENST00000510762.1_1</td>\n",
       "      <td>0.072184</td>\n",
       "      <td>0.236073</td>\n",
       "      <td>0.346634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.090810</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.187659</td>\n",
       "      <td>0.447893</td>\n",
       "      <td>0.050853</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.657279</td>\n",
       "      <td>0.223482</td>\n",
       "      <td>0.245856</td>\n",
       "      <td>0.097768</td>\n",
       "      <td>0.103514</td>\n",
       "      <td>0.438505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111361</th>\n",
       "      <td>ENST00000548706.1_1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.93823</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001041</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.017232</td>\n",
       "      <td>0.690473</td>\n",
       "      <td>0.012292</td>\n",
       "      <td>3.015130</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  target_id  A549_4_1  K562_1_1    H1_1_1  NCI-H460_2_1  \\\n",
       "97645   ENST00000502161.6_1  0.000000  0.000000  0.000000           0.0   \n",
       "165675  ENST00000528059.1_1  0.032773  0.975492  0.362165           0.0   \n",
       "13838   ENST00000255416.8_2  0.000000  0.000000  0.000000           0.0   \n",
       "44299   ENST00000510762.1_1  0.072184  0.236073  0.346634           0.0   \n",
       "111361  ENST00000548706.1_1  0.000000  0.000000  0.000000           0.0   \n",
       "\n",
       "        SK-N-DZ_1_1  SK-MEL-5_1_1  NCI-H460_1_1  SK-N-SH_4_1  HT1080_2_1  \\\n",
       "97645      0.000000       0.00000           0.0     0.022999    0.000000   \n",
       "165675     0.319043       0.00000           0.0     0.029940    0.024811   \n",
       "13838      0.000000       0.00000           0.0     0.015455    0.000000   \n",
       "44299      0.000000       0.00000           0.0     0.090810    0.000000   \n",
       "111361     0.000000       3.93823           0.0     0.001041    0.000000   \n",
       "\n",
       "            ...       GM12878_2_1  SK-N-DZ_2_1  SK-N-SH_3_1  A549_1_1  \\\n",
       "97645       ...          0.073156     0.071750     0.048549       0.0   \n",
       "165675      ...          0.031460     0.000000     0.083639       0.0   \n",
       "13838       ...          0.018436     0.000000     0.043335       0.0   \n",
       "44299       ...          0.187659     0.447893     0.050853       0.0   \n",
       "111361      ...          0.000000     0.000000     0.000000       0.0   \n",
       "\n",
       "        HT1080_1_1  A549_2_1  K562_2_1  GM12878_1_1  IMR-90_1_1  SK-MEL-5_2_1  \n",
       "97645     0.000000  0.000000  0.007496     0.014783    0.000000      0.000000  \n",
       "165675    0.325053  0.000000  0.819158     0.000000    0.008877      0.000000  \n",
       "13838     0.000000  0.000000  0.000000     0.029581    0.000000      0.000000  \n",
       "44299     0.657279  0.223482  0.245856     0.097768    0.103514      0.438505  \n",
       "111361    0.000000  0.000000  0.017232     0.690473    0.012292      3.015130  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transcript_counts = gene_df[gene_df[\"target_id\"].str.startswith(\"ENST\")]\n",
    "print(len(transcript_counts))\n",
    "transcript_counts.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "200140\n"
     ]
    }
   ],
   "source": [
    "transcript_counts = transcript_counts.append(gene_df[gene_df[\"target_id\"] == \"DIGIT\"])\n",
    "print(len(transcript_counts))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. make 1 dataframe with gene/transcript counts mapped to gene ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target_id</th>\n",
       "      <th>A549_4_1</th>\n",
       "      <th>K562_1_1</th>\n",
       "      <th>H1_1_1</th>\n",
       "      <th>NCI-H460_2_1</th>\n",
       "      <th>SK-N-DZ_1_1</th>\n",
       "      <th>SK-MEL-5_1_1</th>\n",
       "      <th>NCI-H460_1_1</th>\n",
       "      <th>SK-N-SH_4_1</th>\n",
       "      <th>HT1080_2_1</th>\n",
       "      <th>...</th>\n",
       "      <th>SK-N-SH_3_1</th>\n",
       "      <th>A549_1_1</th>\n",
       "      <th>HT1080_1_1</th>\n",
       "      <th>A549_2_1</th>\n",
       "      <th>K562_2_1</th>\n",
       "      <th>GM12878_1_1</th>\n",
       "      <th>IMR-90_1_1</th>\n",
       "      <th>SK-MEL-5_2_1</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>200140</th>\n",
       "      <td>ENSG00000223972.5_1::chr1:11868-14409(+)</td>\n",
       "      <td>0.175356</td>\n",
       "      <td>0.152498</td>\n",
       "      <td>0.004712</td>\n",
       "      <td>0.276494</td>\n",
       "      <td>1.34005</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>7.060170e-02</td>\n",
       "      <td>0.136682</td>\n",
       "      <td>0.018867</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.377350e-07</td>\n",
       "      <td>0.086909</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.001038</td>\n",
       "      <td>0.059667</td>\n",
       "      <td>0.402920</td>\n",
       "      <td>ENSG00000223972.5_1</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200141</th>\n",
       "      <td>ENSG00000227232.5_1::chr1:14404-29571(-)</td>\n",
       "      <td>1.476590</td>\n",
       "      <td>1.742750</td>\n",
       "      <td>0.807824</td>\n",
       "      <td>18.798600</td>\n",
       "      <td>19.51290</td>\n",
       "      <td>11.6066</td>\n",
       "      <td>1.747280e+01</td>\n",
       "      <td>8.522320</td>\n",
       "      <td>8.013650</td>\n",
       "      <td>...</td>\n",
       "      <td>6.811900</td>\n",
       "      <td>0.644312</td>\n",
       "      <td>7.090820e+00</td>\n",
       "      <td>0.843098</td>\n",
       "      <td>2.065460</td>\n",
       "      <td>0.531770</td>\n",
       "      <td>1.292600</td>\n",
       "      <td>14.212700</td>\n",
       "      <td>ENSG00000227232.5_1</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200142</th>\n",
       "      <td>ENSG00000243485.4_2::chr1:29553-31109(+)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.080118</td>\n",
       "      <td>0.078441</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>6.189830e-08</td>\n",
       "      <td>0.045724</td>\n",
       "      <td>0.077854</td>\n",
       "      <td>...</td>\n",
       "      <td>0.041129</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.063831</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.689376</td>\n",
       "      <td>ENSG00000243485.4_2</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200143</th>\n",
       "      <td>ENSG00000237613.2_1::chr1:34554-36082(-)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005424</td>\n",
       "      <td>0.000066</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.410140e-03</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.029432</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000237613.2_1</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200144</th>\n",
       "      <td>ENSG00000268020.3_1::chr1:52472-53312(+)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004732</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.011290</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000268020.3_1</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       target_id  A549_4_1  K562_1_1  \\\n",
       "200140  ENSG00000223972.5_1::chr1:11868-14409(+)  0.175356  0.152498   \n",
       "200141  ENSG00000227232.5_1::chr1:14404-29571(-)  1.476590  1.742750   \n",
       "200142  ENSG00000243485.4_2::chr1:29553-31109(+)  0.000000  0.080118   \n",
       "200143  ENSG00000237613.2_1::chr1:34554-36082(-)  0.000000  0.005424   \n",
       "200144  ENSG00000268020.3_1::chr1:52472-53312(+)  0.000000  0.004732   \n",
       "\n",
       "          H1_1_1  NCI-H460_2_1  SK-N-DZ_1_1  SK-MEL-5_1_1  NCI-H460_1_1  \\\n",
       "200140  0.004712      0.276494      1.34005        0.0000  7.060170e-02   \n",
       "200141  0.807824     18.798600     19.51290       11.6066  1.747280e+01   \n",
       "200142  0.078441      0.000000      0.00000        0.0000  6.189830e-08   \n",
       "200143  0.000066      0.000000      0.00000        0.0000  0.000000e+00   \n",
       "200144  0.000000      0.000000      0.00000        0.0000  0.000000e+00   \n",
       "\n",
       "        SK-N-SH_4_1  HT1080_2_1  ...   SK-N-SH_3_1  A549_1_1    HT1080_1_1  \\\n",
       "200140     0.136682    0.018867  ...      0.000000  0.000000  6.377350e-07   \n",
       "200141     8.522320    8.013650  ...      6.811900  0.644312  7.090820e+00   \n",
       "200142     0.045724    0.077854  ...      0.041129  0.000000  0.000000e+00   \n",
       "200143     0.000054    0.000000  ...      0.000000  0.000000  1.410140e-03   \n",
       "200144     0.000000    0.000000  ...      0.000000  0.000000  0.000000e+00   \n",
       "\n",
       "        A549_2_1  K562_2_1  GM12878_1_1  IMR-90_1_1  SK-MEL-5_2_1  \\\n",
       "200140  0.086909  0.000024     0.001038    0.059667      0.402920   \n",
       "200141  0.843098  2.065460     0.531770    1.292600     14.212700   \n",
       "200142  0.000000  0.063831     0.000000    0.000000      0.689376   \n",
       "200143  0.000000  0.029432     0.000000    0.000105      0.000000   \n",
       "200144  0.000000  0.011290     0.000000    0.000000      0.000000   \n",
       "\n",
       "                    gene_id  type  \n",
       "200140  ENSG00000223972.5_1  gene  \n",
       "200141  ENSG00000227232.5_1  gene  \n",
       "200142  ENSG00000243485.4_2  gene  \n",
       "200143  ENSG00000237613.2_1  gene  \n",
       "200144  ENSG00000268020.3_1  gene  \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_counts[\"gene_id\"] = gene_counts[\"target_id\"].str.split(\"::\", expand=True)[0]\n",
    "gene_counts[\"type\"] = \"gene\"\n",
    "gene_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target_id</th>\n",
       "      <th>A549_4_1</th>\n",
       "      <th>K562_1_1</th>\n",
       "      <th>H1_1_1</th>\n",
       "      <th>NCI-H460_2_1</th>\n",
       "      <th>SK-N-DZ_1_1</th>\n",
       "      <th>SK-MEL-5_1_1</th>\n",
       "      <th>NCI-H460_1_1</th>\n",
       "      <th>SK-N-SH_4_1</th>\n",
       "      <th>HT1080_2_1</th>\n",
       "      <th>...</th>\n",
       "      <th>SK-N-SH_3_1</th>\n",
       "      <th>A549_1_1</th>\n",
       "      <th>HT1080_1_1</th>\n",
       "      <th>A549_2_1</th>\n",
       "      <th>K562_2_1</th>\n",
       "      <th>GM12878_1_1</th>\n",
       "      <th>IMR-90_1_1</th>\n",
       "      <th>SK-MEL-5_2_1</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000456328.2_1</td>\n",
       "      <td>0.275527</td>\n",
       "      <td>0.084567</td>\n",
       "      <td>0.000879</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.001765</td>\n",
       "      <td>1.444350</td>\n",
       "      <td>0.733059</td>\n",
       "      <td>0.404552</td>\n",
       "      <td>0.262934</td>\n",
       "      <td>...</td>\n",
       "      <td>0.220057</td>\n",
       "      <td>0.359035</td>\n",
       "      <td>0.269616</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.093201</td>\n",
       "      <td>0.013761</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.135529</td>\n",
       "      <td>ENST00000456328.2_1</td>\n",
       "      <td>transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000450305.2_1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENST00000450305.2_1</td>\n",
       "      <td>transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000488147.1_1</td>\n",
       "      <td>14.017000</td>\n",
       "      <td>14.784400</td>\n",
       "      <td>5.105190</td>\n",
       "      <td>56.5208</td>\n",
       "      <td>88.337500</td>\n",
       "      <td>49.514600</td>\n",
       "      <td>62.705200</td>\n",
       "      <td>22.380200</td>\n",
       "      <td>28.958800</td>\n",
       "      <td>...</td>\n",
       "      <td>20.399800</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>28.861400</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.014800</td>\n",
       "      <td>3.172690</td>\n",
       "      <td>16.9903</td>\n",
       "      <td>77.177500</td>\n",
       "      <td>ENST00000488147.1_1</td>\n",
       "      <td>transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.429972</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.036563</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.028681</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.037715</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000469289.1_1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.048793</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.189307</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.297624</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENST00000469289.1_1</td>\n",
       "      <td>transcript</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             target_id   A549_4_1   K562_1_1    H1_1_1  NCI-H460_2_1  \\\n",
       "0  ENST00000456328.2_1   0.275527   0.084567  0.000879        0.0000   \n",
       "1  ENST00000450305.2_1   0.000000   0.000000  0.000000        0.0000   \n",
       "2  ENST00000488147.1_1  14.017000  14.784400  5.105190       56.5208   \n",
       "3  ENST00000473358.1_1   0.000000   0.000000  0.000000        0.0000   \n",
       "4  ENST00000469289.1_1   0.000000   0.048793  0.000000        0.0000   \n",
       "\n",
       "   SK-N-DZ_1_1  SK-MEL-5_1_1  NCI-H460_1_1  SK-N-SH_4_1  HT1080_2_1  \\\n",
       "0     0.001765      1.444350      0.733059     0.404552    0.262934   \n",
       "1     0.000000      0.000000      0.000000     0.000000    0.000000   \n",
       "2    88.337500     49.514600     62.705200    22.380200   28.958800   \n",
       "3     0.000000      0.429972      0.000000     0.036563    0.000000   \n",
       "4     0.000000      0.000000      0.000000     0.189307    0.000000   \n",
       "\n",
       "      ...      SK-N-SH_3_1  A549_1_1  HT1080_1_1  A549_2_1   K562_2_1  \\\n",
       "0     ...         0.220057  0.359035    0.269616       0.0   0.093201   \n",
       "1     ...         0.000000  0.000000    0.000000       0.0   0.000000   \n",
       "2     ...        20.399800  0.000000   28.861400       0.0  15.014800   \n",
       "3     ...         0.028681  0.000000    0.000000       0.0   0.037715   \n",
       "4     ...         0.297624  0.000000    0.000000       0.0   0.000000   \n",
       "\n",
       "   GM12878_1_1  IMR-90_1_1  SK-MEL-5_2_1        transcript_id        type  \n",
       "0     0.013761      0.0000      0.135529  ENST00000456328.2_1  transcript  \n",
       "1     0.000000      0.0000      0.000000  ENST00000450305.2_1  transcript  \n",
       "2     3.172690     16.9903     77.177500  ENST00000488147.1_1  transcript  \n",
       "3     0.000000      0.0000      0.000000  ENST00000473358.1_1  transcript  \n",
       "4     0.000000      0.0000      0.000000  ENST00000469289.1_1  transcript  \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transcript_counts[\"transcript_id\"] = transcript_counts[\"target_id\"]\n",
    "transcript_counts[\"type\"] = \"transcript\"\n",
    "transcript_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cols = [\"index\", \"type\", \"hESC_rep1\", \"hESC_rep2\", \"endo_rep1\", \"endo_rep2\", \"meso_rep1\", \"meso_rep2\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A549_1_1</th>\n",
       "      <th>A549_2_1</th>\n",
       "      <th>A549_3_1</th>\n",
       "      <th>A549_4_1</th>\n",
       "      <th>GM12878_1_1</th>\n",
       "      <th>GM12878_2_1</th>\n",
       "      <th>H1_1_1</th>\n",
       "      <th>HT1080_1_1</th>\n",
       "      <th>HT1080_2_1</th>\n",
       "      <th>HUES64_1_3</th>\n",
       "      <th>...</th>\n",
       "      <th>NCI-H460_2_1</th>\n",
       "      <th>SK-MEL-5_1_1</th>\n",
       "      <th>SK-MEL-5_2_1</th>\n",
       "      <th>SK-N-DZ_1_1</th>\n",
       "      <th>SK-N-DZ_2_1</th>\n",
       "      <th>SK-N-SH_3_1</th>\n",
       "      <th>SK-N-SH_4_1</th>\n",
       "      <th>target_id</th>\n",
       "      <th>type</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>250579</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>ENSG00000238705.1::chr1:26881032-26881084(+)</td>\n",
       "      <td>gene</td>\n",
       "      <td>ENSG00000238705.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248007</th>\n",
       "      <td>0.001523</td>\n",
       "      <td>0.001126</td>\n",
       "      <td>0.003038</td>\n",
       "      <td>0.002922</td>\n",
       "      <td>0.002244</td>\n",
       "      <td>0.001904</td>\n",
       "      <td>0.009758</td>\n",
       "      <td>0.014438</td>\n",
       "      <td>0.021905</td>\n",
       "      <td>1.516150e-03</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002794</td>\n",
       "      <td>0.02297</td>\n",
       "      <td>2.055250e-02</td>\n",
       "      <td>0.007273</td>\n",
       "      <td>0.000840</td>\n",
       "      <td>0.000580</td>\n",
       "      <td>0.00130</td>\n",
       "      <td>ENSG00000146950.12_1::chrX:9754495-9917483(+)</td>\n",
       "      <td>gene</td>\n",
       "      <td>ENSG00000146950.12_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16903</th>\n",
       "      <td>79.887500</td>\n",
       "      <td>99.212900</td>\n",
       "      <td>79.239300</td>\n",
       "      <td>119.758000</td>\n",
       "      <td>33.313400</td>\n",
       "      <td>37.443400</td>\n",
       "      <td>10.243900</td>\n",
       "      <td>4.803880</td>\n",
       "      <td>5.922840</td>\n",
       "      <td>6.062730e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>28.982500</td>\n",
       "      <td>27.13440</td>\n",
       "      <td>3.068600e+01</td>\n",
       "      <td>75.665300</td>\n",
       "      <td>70.891000</td>\n",
       "      <td>36.301200</td>\n",
       "      <td>39.96750</td>\n",
       "      <td>ENST00000396290.1_1</td>\n",
       "      <td>transcript</td>\n",
       "      <td>ENST00000396290.1_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82465</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.340984</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.285105</td>\n",
       "      <td>0.454248</td>\n",
       "      <td>0.053929</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>ENST00000525376.1_1</td>\n",
       "      <td>transcript</td>\n",
       "      <td>ENST00000525376.1_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111137</th>\n",
       "      <td>12.717100</td>\n",
       "      <td>15.451100</td>\n",
       "      <td>45.709600</td>\n",
       "      <td>42.476500</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>12.469900</td>\n",
       "      <td>12.333800</td>\n",
       "      <td>6.753570e-09</td>\n",
       "      <td>...</td>\n",
       "      <td>1.535850</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>2.954430e-08</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.127550</td>\n",
       "      <td>6.79547</td>\n",
       "      <td>ENST00000313234.9_1</td>\n",
       "      <td>transcript</td>\n",
       "      <td>ENST00000313234.9_1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         A549_1_1   A549_2_1   A549_3_1    A549_4_1  GM12878_1_1  GM12878_2_1  \\\n",
       "250579   0.000000   0.000000   0.000000    0.000000     0.000000     0.000000   \n",
       "248007   0.001523   0.001126   0.003038    0.002922     0.002244     0.001904   \n",
       "16903   79.887500  99.212900  79.239300  119.758000    33.313400    37.443400   \n",
       "82465    0.000000   0.000000   0.000000    0.000000     0.000000     0.000000   \n",
       "111137  12.717100  15.451100  45.709600   42.476500     0.000000     0.000000   \n",
       "\n",
       "           H1_1_1  HT1080_1_1  HT1080_2_1    HUES64_1_3          ...           \\\n",
       "250579   0.000000    0.000000    0.000000  0.000000e+00          ...            \n",
       "248007   0.009758    0.014438    0.021905  1.516150e-03          ...            \n",
       "16903   10.243900    4.803880    5.922840  6.062730e+00          ...            \n",
       "82465    0.340984    0.000000    0.000000  0.000000e+00          ...            \n",
       "111137   0.000000   12.469900   12.333800  6.753570e-09          ...            \n",
       "\n",
       "        NCI-H460_2_1  SK-MEL-5_1_1  SK-MEL-5_2_1  SK-N-DZ_1_1  SK-N-DZ_2_1  \\\n",
       "250579      0.000000       0.00000  0.000000e+00     0.000000     0.000000   \n",
       "248007      0.002794       0.02297  2.055250e-02     0.007273     0.000840   \n",
       "16903      28.982500      27.13440  3.068600e+01    75.665300    70.891000   \n",
       "82465       0.000000       0.00000  0.000000e+00     0.285105     0.454248   \n",
       "111137      1.535850       0.00000  2.954430e-08     0.000000     0.000000   \n",
       "\n",
       "        SK-N-SH_3_1  SK-N-SH_4_1  \\\n",
       "250579     0.000000      0.00000   \n",
       "248007     0.000580      0.00130   \n",
       "16903     36.301200     39.96750   \n",
       "82465      0.053929      0.00000   \n",
       "111137     6.127550      6.79547   \n",
       "\n",
       "                                            target_id        type  \\\n",
       "250579   ENSG00000238705.1::chr1:26881032-26881084(+)        gene   \n",
       "248007  ENSG00000146950.12_1::chrX:9754495-9917483(+)        gene   \n",
       "16903                             ENST00000396290.1_1  transcript   \n",
       "82465                             ENST00000525376.1_1  transcript   \n",
       "111137                            ENST00000313234.9_1  transcript   \n",
       "\n",
       "                          id  \n",
       "250579     ENSG00000238705.1  \n",
       "248007  ENSG00000146950.12_1  \n",
       "16903    ENST00000396290.1_1  \n",
       "82465    ENST00000525376.1_1  \n",
       "111137   ENST00000313234.9_1  \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tot_counts = gene_counts.append(transcript_counts)\n",
    "tot_counts[\"id\"] = tot_counts[\"target_id\"].str.split(\"::\", expand=True)[0]\n",
    "tot_counts.drop([\"gene_id\", \"transcript_id\"], axis=1, inplace=True)\n",
    "tot_counts.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tot_counts = tot_counts.merge(gene_map, left_on=\"id\", right_on=\"transcript_id\", how=\"left\")\n",
    "tot_counts.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fill_gene_id(row):\n",
    "    if pd.isnull(row[\"gene_id\"]):\n",
    "        return row[\"id\"]\n",
    "    else:\n",
    "        return row[\"gene_id\"]\n",
    "    \n",
    "tot_counts[\"gene_id\"] = tot_counts.apply(fill_gene_id, axis=1)\n",
    "tot_counts.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"id\", \"gene_id\", \"type\"]\n",
    "cols.extend(samps)\n",
    "\n",
    "tot_counts = tot_counts[cols]\n",
    "tot_counts = tot_counts.sort_values(by=\"gene_id\")\n",
    "tot_counts.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. sum counts per gene (incl. gene isoform and all transcripts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sum_counts = tot_counts.groupby(\"gene_id\")[list(samps)].agg(\"sum\").reset_index()\n",
    "sum_counts.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. calculate splicing efficiency"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(gene_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(sum_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = gene_counts.merge(sum_counts, on=\"gene_id\", suffixes=(\"_gene\", \"_tot\"))\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for samp in samps:\n",
    "    data[\"%s_eff\" % samp] = 1-((data[\"%s_gene\" % samp]+1)/(data[\"%s_tot\" % samp]+1))\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cells = set([x.split(\"_\")[0] for x in samps])\n",
    "cells"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for cell in cells:\n",
    "    sub_cols = [\"%s_eff\" % x for x in samps if cell in x]\n",
    "    print(\"cell: %s, samps: %s\" % (cell, sub_cols))\n",
    "    data[\"%s_eff_mean\" % cell] = data[sub_cols].mean(axis=1)\n",
    "    sub_cols = [\"%s_tot\" % x for x in samps if cell in x]\n",
    "    data[\"%s_exp_mean\" % cell] = data[sub_cols].mean(axis=1)\n",
    "    \n",
    "    # also record the ratio of expr & splicing efficiency (eff * expr)\n",
    "    data[\"%s_eff_ratio\" % cell] = data[\"%s_eff_mean\" % cell] * data[\"%s_exp_mean\" % cell]\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data[\"gene_id\"] = data.target_id.str.split(\"::\", expand=True)[0]\n",
    "\n",
    "clean_cols = [\"gene_id\"]\n",
    "clean_cols.extend([x for x in data.columns if \"_mean\" in x or \"_ratio\" in x])\n",
    "data = data[clean_cols]\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. merge w/ biotype info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "biotypes = rna_seq[[\"gene_id\", \"gene_name\", \"csf\", \"cleaner_gene_biotype\"]].drop_duplicates()\n",
    "print(len(biotypes))\n",
    "biotypes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = data.merge(biotypes, on=\"gene_id\", how=\"left\")\n",
    "print(len(data))\n",
    "data.sample(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 9. some QC plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.cleaner_gene_biotype.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "order = [\"protein_coding\", \"intergenic\", \"promoter_overlap\", \"transcript_overlap\", \"gene_nearby\"]\n",
    "pal = {\"protein_coding\": sns.color_palette(\"deep\")[0], \"intergenic\": \"firebrick\", \"promoter_overlap\": \"firebrick\",\n",
    "       \"transcript_overlap\": \"firebrick\", \"gene_nearby\": \"firebrick\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sorted(list(cells))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, axarr = plt.subplots(figsize=(6, 4), nrows=4, ncols=3, sharex=True)\n",
    "\n",
    "c = 0\n",
    "for i in range(4):\n",
    "    for j in range(3):\n",
    "        col = sorted(list(cells))[c]\n",
    "\n",
    "        sub = data[data[\"%s_exp_mean\" % col] >= 0.1]\n",
    "        ax = axarr[i, j]\n",
    "\n",
    "        sns.boxplot(data=sub, x=\"cleaner_gene_biotype\", y=\"%s_eff_mean\" % col, order=order, palette=pal,\n",
    "                    flierprops=dict(marker='o', markersize=3), ax=ax)\n",
    "        mimic_r_boxplot(ax)\n",
    "\n",
    "        ax.set_xlabel(\"\")\n",
    "        ax.set_ylabel(col)\n",
    "        ax.set_xticklabels([\"protein-coding\", \"intergenic\", \"promoter overlap\", \"transcript overlap\", \"gene nearby\"],\n",
    "                           rotation=50, ha=\"right\", va=\"top\")\n",
    "        c += 1\n",
    "\n",
    "plt.subplots_adjust(wspace=0.4)\n",
    "plt.text(0.05, 0.5, 'splicing efficiency in:',\n",
    "         horizontalalignment='right',\n",
    "         verticalalignment='center',\n",
    "         rotation='vertical',\n",
    "         transform=plt.gcf().transFigure)\n",
    "\n",
    "fig.savefig(\"splicing_eff_boxplot.pdf\", dpi=\"figure\", bbox_inches=\"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "eff_cols = [x for x in data.columns if \"eff\" in x and \"ratio\" not in x]\n",
    "exp_cols = [x for x in data.columns if \"exp\" in x]\n",
    "ratio_cols = [x for x in data.columns if \"ratio\" in x]\n",
    "\n",
    "data[\"max_eff\"] = data[eff_cols].max(axis=1)\n",
    "data[\"max_exp\"] = data[exp_cols].max(axis=1)\n",
    "data[\"max_ratio\"] = data[ratio_cols].max(axis=1)\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 10. write file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_f = \"../../../misc/08__model_features/gene_splicing_efficiency.with_DIGIT.txt\"\n",
    "data.to_csv(data_f, index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:.conda-py3]",
   "language": "python",
   "name": "conda-env-.conda-py3-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
