{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 03__hit_gwas\n",
    "\n",
    "in this notebook, i examine whether our hits have GWAS-associated SNPs in close proximity\n",
    "\n",
    "figures in this notebook:\n",
    "- Fig 5C: barplot showing % of hits vs. non-hits with an endo-cancer associated SNP within 5kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import math\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import re\n",
    "import scipy.stats as stats\n",
    "import seaborn as sns\n",
    "import sys\n",
    "import time\n",
    "\n",
    "from scipy.stats import spearmanr\n",
    "from scipy.stats import pearsonr\n",
    "from statsmodels.sandbox.stats import multicomp\n",
    "\n",
    "# import utils\n",
    "sys.path.append(\"../../../utils\")\n",
    "from plotting_utils import *\n",
    "\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'svg'\n",
    "mpl.rcParams['figure.autolayout'] = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.set(**PAPER_PRESET)\n",
    "fontsize = PAPER_FONTSIZE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "index_f = \"../../../data/01__design/02__final_sgRNAs/crispri_picked_sgRNAs.not_deduped.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_f = \"../../../data/02__screen/02__enrichment_data/enrichment_values.with_rna_seq.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# all gwas files\n",
    "gwas_dir = \"../../../misc/06__gwas\"\n",
    "closest_gwas_f = \"%s/tss_coords.closest_gwas_snp.with_DIGIT.bed\" % gwas_dir\n",
    "closest_cancer_f = \"%s/tss_coords.closest_all_cancer_snp.with_DIGIT.bed\" % gwas_dir\n",
    "closest_endo_f = \"%s/tss_coords.closest_endo_cancer_snp.with_DIGIT.bed\" % gwas_dir\n",
    "closest_gwas_gb_f = \"%s/transcript_coords.closest_gwas_snp.with_DIGIT.bed\" % gwas_dir\n",
    "closest_cancer_gb_f = \"%s/transcript_coords.closest_all_cancer_snp.with_DIGIT.bed\" % gwas_dir\n",
    "closest_endo_gb_f = \"%s/transcript_coords.closest_endo_cancer_snp.with_DIGIT.bed\" % gwas_dir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>tss_type</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>sgRNA_qual</th>\n",
       "      <th>cut_offset</th>\n",
       "      <th>rank</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTACCTTGGAGAAGAAATGG</td>\n",
       "      <td>NC_000001.11:+:100894913</td>\n",
       "      <td>ENST00000609247.1_1</td>\n",
       "      <td>RP4-549L20.3-001</td>\n",
       "      <td>ENSG00000273204.1_1</td>\n",
       "      <td>RP4-549L20.3</td>\n",
       "      <td>FANTOM CAT assignment</td>\n",
       "      <td>chr1:101360463..101360478,+</td>\n",
       "      <td>good</td>\n",
       "      <td>42.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATGTTTAAAAGTTGTTACCT</td>\n",
       "      <td>NC_000001.11:+:100894913</td>\n",
       "      <td>ENST00000609247.1_1</td>\n",
       "      <td>RP4-549L20.3-001</td>\n",
       "      <td>ENSG00000273204.1_1</td>\n",
       "      <td>RP4-549L20.3</td>\n",
       "      <td>FANTOM CAT assignment</td>\n",
       "      <td>chr1:101360463..101360478,+</td>\n",
       "      <td>good</td>\n",
       "      <td>56.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TTTCCTCCATTTCTTCTCCA</td>\n",
       "      <td>NC_000001.11:+:100894913</td>\n",
       "      <td>ENST00000609247.1_1</td>\n",
       "      <td>RP4-549L20.3-001</td>\n",
       "      <td>ENSG00000273204.1_1</td>\n",
       "      <td>RP4-549L20.3</td>\n",
       "      <td>FANTOM CAT assignment</td>\n",
       "      <td>chr1:101360463..101360478,+</td>\n",
       "      <td>good</td>\n",
       "      <td>50.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GAAGAAATGGAGGAAAGGAT</td>\n",
       "      <td>NC_000001.11:+:100894913</td>\n",
       "      <td>ENST00000609247.1_1</td>\n",
       "      <td>RP4-549L20.3-001</td>\n",
       "      <td>ENSG00000273204.1_1</td>\n",
       "      <td>RP4-549L20.3</td>\n",
       "      <td>FANTOM CAT assignment</td>\n",
       "      <td>chr1:101360463..101360478,+</td>\n",
       "      <td>good</td>\n",
       "      <td>32.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TTGGAGAAGAAATGGAGGAA</td>\n",
       "      <td>NC_000001.11:+:100894913</td>\n",
       "      <td>ENST00000609247.1_1</td>\n",
       "      <td>RP4-549L20.3-001</td>\n",
       "      <td>ENSG00000273204.1_1</td>\n",
       "      <td>RP4-549L20.3</td>\n",
       "      <td>FANTOM CAT assignment</td>\n",
       "      <td>chr1:101360463..101360478,+</td>\n",
       "      <td>good</td>\n",
       "      <td>37.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  sgRNA                    tss_id        transcript_id  \\\n",
       "0  TTACCTTGGAGAAGAAATGG  NC_000001.11:+:100894913  ENST00000609247.1_1   \n",
       "1  ATGTTTAAAAGTTGTTACCT  NC_000001.11:+:100894913  ENST00000609247.1_1   \n",
       "2  TTTCCTCCATTTCTTCTCCA  NC_000001.11:+:100894913  ENST00000609247.1_1   \n",
       "3  GAAGAAATGGAGGAAAGGAT  NC_000001.11:+:100894913  ENST00000609247.1_1   \n",
       "4  TTGGAGAAGAAATGGAGGAA  NC_000001.11:+:100894913  ENST00000609247.1_1   \n",
       "\n",
       "    transcript_name              gene_id     gene_name               tss_type  \\\n",
       "0  RP4-549L20.3-001  ENSG00000273204.1_1  RP4-549L20.3  FANTOM CAT assignment   \n",
       "1  RP4-549L20.3-001  ENSG00000273204.1_1  RP4-549L20.3  FANTOM CAT assignment   \n",
       "2  RP4-549L20.3-001  ENSG00000273204.1_1  RP4-549L20.3  FANTOM CAT assignment   \n",
       "3  RP4-549L20.3-001  ENSG00000273204.1_1  RP4-549L20.3  FANTOM CAT assignment   \n",
       "4  RP4-549L20.3-001  ENSG00000273204.1_1  RP4-549L20.3  FANTOM CAT assignment   \n",
       "\n",
       "                       cage_id sgRNA_qual  cut_offset  rank  \n",
       "0  chr1:101360463..101360478,+       good        42.0   1.0  \n",
       "1  chr1:101360463..101360478,+       good        56.0   2.0  \n",
       "2  chr1:101360463..101360478,+       good        50.0   3.0  \n",
       "3  chr1:101360463..101360478,+       good        32.0   4.0  \n",
       "4  chr1:101360463..101360478,+       good        37.0   5.0  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index = pd.read_table(index_f)\n",
    "index.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group_id</th>\n",
       "      <th>ctrl_status</th>\n",
       "      <th>endo_ctrl_val</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>cleaner_transcript_biotype</th>\n",
       "      <th>cleaner_gene_biotype</th>\n",
       "      <th>BFP+_score__rep1</th>\n",
       "      <th>...</th>\n",
       "      <th>combined_pval</th>\n",
       "      <th>padj</th>\n",
       "      <th>neg_log_padj</th>\n",
       "      <th>is_hit</th>\n",
       "      <th>csf</th>\n",
       "      <th>hESC_mean</th>\n",
       "      <th>endo_mean</th>\n",
       "      <th>qval_hESC_endo</th>\n",
       "      <th>endo_hESC_log2fc</th>\n",
       "      <th>endo_hESC_abslog2fc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000008.11:+:54457934</td>\n",
       "      <td>control</td>\n",
       "      <td>True</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>ENSG00000164736.5_1</td>\n",
       "      <td>SOX17-001</td>\n",
       "      <td>ENST00000297316.4_1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>232.279434</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.000284</td>\n",
       "      <td>3.546586</td>\n",
       "      <td>stringent hit</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.110290</td>\n",
       "      <td>134.426513</td>\n",
       "      <td>0.000991006472787698</td>\n",
       "      <td>6.930430</td>\n",
       "      <td>6.930430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000003.12:-:27722707</td>\n",
       "      <td>control</td>\n",
       "      <td>True</td>\n",
       "      <td>EOMES</td>\n",
       "      <td>ENSG00000163508.12_1</td>\n",
       "      <td>EOMES-004</td>\n",
       "      <td>ENST00000461503.2_1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>156.570722</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.000284</td>\n",
       "      <td>3.546586</td>\n",
       "      <td>stringent hit</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.123413</td>\n",
       "      <td>101.207791</td>\n",
       "      <td>0.0217931946808208</td>\n",
       "      <td>6.507473</td>\n",
       "      <td>6.507473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000003.12:-:27722322</td>\n",
       "      <td>control</td>\n",
       "      <td>True</td>\n",
       "      <td>EOMES</td>\n",
       "      <td>ENSG00000163508.12_1</td>\n",
       "      <td>EOMES-001</td>\n",
       "      <td>ENST00000295743.8_1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>43.594914</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.000284</td>\n",
       "      <td>3.546586</td>\n",
       "      <td>stringent hit</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.084601</td>\n",
       "      <td>85.188250</td>\n",
       "      <td>0.000510199113682609</td>\n",
       "      <td>6.312255</td>\n",
       "      <td>6.312255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000016.10:+:23452748</td>\n",
       "      <td>experimental</td>\n",
       "      <td>False</td>\n",
       "      <td>CTD-2270L9.4</td>\n",
       "      <td>ENSG00000260136.5_1</td>\n",
       "      <td>CTD-2270L9.4-001</td>\n",
       "      <td>ENST00000570080.1_1</td>\n",
       "      <td>promoter_overlap</td>\n",
       "      <td>promoter_overlap</td>\n",
       "      <td>3.702626</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000173</td>\n",
       "      <td>0.009619</td>\n",
       "      <td>2.016871</td>\n",
       "      <td>stringent hit</td>\n",
       "      <td>lncRNA_good_csf</td>\n",
       "      <td>1.391981</td>\n",
       "      <td>2.482487</td>\n",
       "      <td>0.0347336116955254</td>\n",
       "      <td>0.541912</td>\n",
       "      <td>0.541912</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000016.10:+:23452748</td>\n",
       "      <td>experimental</td>\n",
       "      <td>False</td>\n",
       "      <td>CTD-2270L9.4</td>\n",
       "      <td>ENSG00000260136.5_1</td>\n",
       "      <td>CTD-2270L9.4-002</td>\n",
       "      <td>ENST00000562842.1_1</td>\n",
       "      <td>promoter_overlap</td>\n",
       "      <td>promoter_overlap</td>\n",
       "      <td>3.702626</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000173</td>\n",
       "      <td>0.009619</td>\n",
       "      <td>2.016871</td>\n",
       "      <td>stringent hit</td>\n",
       "      <td>lncRNA_good_csf</td>\n",
       "      <td>0.602179</td>\n",
       "      <td>1.020428</td>\n",
       "      <td>0.191063464740555</td>\n",
       "      <td>0.334626</td>\n",
       "      <td>0.334626</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  group_id   ctrl_status  endo_ctrl_val     gene_name  \\\n",
       "0  NC_000008.11:+:54457934       control           True         SOX17   \n",
       "1  NC_000003.12:-:27722707       control           True         EOMES   \n",
       "2  NC_000003.12:-:27722322       control           True         EOMES   \n",
       "3  NC_000016.10:+:23452748  experimental          False  CTD-2270L9.4   \n",
       "4  NC_000016.10:+:23452748  experimental          False  CTD-2270L9.4   \n",
       "\n",
       "                gene_id   transcript_name        transcript_id  \\\n",
       "0   ENSG00000164736.5_1         SOX17-001  ENST00000297316.4_1   \n",
       "1  ENSG00000163508.12_1         EOMES-004  ENST00000461503.2_1   \n",
       "2  ENSG00000163508.12_1         EOMES-001  ENST00000295743.8_1   \n",
       "3   ENSG00000260136.5_1  CTD-2270L9.4-001  ENST00000570080.1_1   \n",
       "4   ENSG00000260136.5_1  CTD-2270L9.4-002  ENST00000562842.1_1   \n",
       "\n",
       "  cleaner_transcript_biotype cleaner_gene_biotype  BFP+_score__rep1  \\\n",
       "0             protein_coding       protein_coding        232.279434   \n",
       "1             protein_coding       protein_coding        156.570722   \n",
       "2             protein_coding       protein_coding         43.594914   \n",
       "3           promoter_overlap     promoter_overlap          3.702626   \n",
       "4           promoter_overlap     promoter_overlap          3.702626   \n",
       "\n",
       "          ...           combined_pval      padj  neg_log_padj         is_hit  \\\n",
       "0         ...                0.000002  0.000284      3.546586  stringent hit   \n",
       "1         ...                0.000002  0.000284      3.546586  stringent hit   \n",
       "2         ...                0.000002  0.000284      3.546586  stringent hit   \n",
       "3         ...                0.000173  0.009619      2.016871  stringent hit   \n",
       "4         ...                0.000173  0.009619      2.016871  stringent hit   \n",
       "\n",
       "               csf  hESC_mean   endo_mean        qval_hESC_endo  \\\n",
       "0   protein_coding   0.110290  134.426513  0.000991006472787698   \n",
       "1   protein_coding   0.123413  101.207791    0.0217931946808208   \n",
       "2   protein_coding   0.084601   85.188250  0.000510199113682609   \n",
       "3  lncRNA_good_csf   1.391981    2.482487    0.0347336116955254   \n",
       "4  lncRNA_good_csf   0.602179    1.020428     0.191063464740555   \n",
       "\n",
       "   endo_hESC_log2fc  endo_hESC_abslog2fc  \n",
       "0          6.930430             6.930430  \n",
       "1          6.507473             6.507473  \n",
       "2          6.312255             6.312255  \n",
       "3          0.541912             0.541912  \n",
       "4          0.334626             0.334626  \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_table(data_f)\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>snp_chr</th>\n",
       "      <th>snp_start</th>\n",
       "      <th>snp_end</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>29555</td>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>chr1</td>\n",
       "      <td>780396</td>\n",
       "      <td>780397</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>750842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>36072</td>\n",
       "      <td>36073</td>\n",
       "      <td>ENST00000461467.1_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>780396</td>\n",
       "      <td>780397</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>744324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>91104</td>\n",
       "      <td>91105</td>\n",
       "      <td>ENST00000495576.1_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>780396</td>\n",
       "      <td>780397</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>689292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120931</td>\n",
       "      <td>120932</td>\n",
       "      <td>ENST00000466430.5_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>780396</td>\n",
       "      <td>780397</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>659465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>133565</td>\n",
       "      <td>133566</td>\n",
       "      <td>ENST00000453576.2_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>780396</td>\n",
       "      <td>780397</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>646831</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr   start     end        transcript_id  score strand snp_chr  snp_start  \\\n",
       "0  chr1   29554   29555  ENST00000473358.1_1      0      +    chr1     780396   \n",
       "1  chr1   36072   36073  ENST00000461467.1_1      0      -    chr1     780396   \n",
       "2  chr1   91104   91105  ENST00000495576.1_1      0      -    chr1     780396   \n",
       "3  chr1  120931  120932  ENST00000466430.5_1      0      -    chr1     780396   \n",
       "4  chr1  133565  133566  ENST00000453576.2_1      0      -    chr1     780396   \n",
       "\n",
       "   snp_end       snp_id           disease  distance  \n",
       "0   780397  rs141175086  circadian rhythm    750842  \n",
       "1   780397  rs141175086  circadian rhythm    744324  \n",
       "2   780397  rs141175086  circadian rhythm    689292  \n",
       "3   780397  rs141175086  circadian rhythm    659465  \n",
       "4   780397  rs141175086  circadian rhythm    646831  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "closest_gwas = pd.read_table(closest_gwas_f, sep=\"\\t\", header=None)\n",
    "closest_gwas.columns = [\"chr\", \"start\", \"end\", \"transcript_id\", \"score\", \"strand\", \"snp_chr\", \"snp_start\", \"snp_end\",\n",
    "                        \"snp_id\", \"disease\", \"distance\"]\n",
    "closest_gwas.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>snp_chr</th>\n",
       "      <th>snp_start</th>\n",
       "      <th>snp_end</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>29555</td>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>36072</td>\n",
       "      <td>36073</td>\n",
       "      <td>ENST00000461467.1_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>91104</td>\n",
       "      <td>91105</td>\n",
       "      <td>ENST00000495576.1_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120931</td>\n",
       "      <td>120932</td>\n",
       "      <td>ENST00000466430.5_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>133565</td>\n",
       "      <td>133566</td>\n",
       "      <td>ENST00000453576.2_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr   start     end        transcript_id  score strand snp_chr  snp_start  \\\n",
       "0  chr1   29554   29555  ENST00000473358.1_1      0      +    chr1     894572   \n",
       "1  chr1   36072   36073  ENST00000461467.1_1      0      -    chr1     894572   \n",
       "2  chr1   91104   91105  ENST00000495576.1_1      0      -    chr1     894572   \n",
       "3  chr1  120931  120932  ENST00000466430.5_1      0      -    chr1     894572   \n",
       "4  chr1  133565  133566  ENST00000453576.2_1      0      -    chr1     894572   \n",
       "\n",
       "   snp_end      snp_id               disease  distance  \n",
       "0   894573  rs13303010  pancreatic carcinoma    865018  \n",
       "1   894573  rs13303010  pancreatic carcinoma    858500  \n",
       "2   894573  rs13303010  pancreatic carcinoma    803468  \n",
       "3   894573  rs13303010  pancreatic carcinoma    773641  \n",
       "4   894573  rs13303010  pancreatic carcinoma    761007  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "closest_cancer = pd.read_table(closest_cancer_f, sep=\"\\t\", header=None)\n",
    "closest_cancer.columns = [\"chr\", \"start\", \"end\", \"transcript_id\", \"score\", \"strand\", \"snp_chr\", \"snp_start\", \"snp_end\",\n",
    "                        \"snp_id\", \"disease\", \"distance\"]\n",
    "closest_cancer.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>snp_chr</th>\n",
       "      <th>snp_start</th>\n",
       "      <th>snp_end</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>29555</td>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>36072</td>\n",
       "      <td>36073</td>\n",
       "      <td>ENST00000461467.1_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>91104</td>\n",
       "      <td>91105</td>\n",
       "      <td>ENST00000495576.1_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120931</td>\n",
       "      <td>120932</td>\n",
       "      <td>ENST00000466430.5_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>133565</td>\n",
       "      <td>133566</td>\n",
       "      <td>ENST00000453576.2_1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr   start     end        transcript_id  score strand snp_chr  snp_start  \\\n",
       "0  chr1   29554   29555  ENST00000473358.1_1      0      +    chr1     894572   \n",
       "1  chr1   36072   36073  ENST00000461467.1_1      0      -    chr1     894572   \n",
       "2  chr1   91104   91105  ENST00000495576.1_1      0      -    chr1     894572   \n",
       "3  chr1  120931  120932  ENST00000466430.5_1      0      -    chr1     894572   \n",
       "4  chr1  133565  133566  ENST00000453576.2_1      0      -    chr1     894572   \n",
       "\n",
       "   snp_end      snp_id               disease  distance  \n",
       "0   894573  rs13303010  pancreatic carcinoma    865018  \n",
       "1   894573  rs13303010  pancreatic carcinoma    858500  \n",
       "2   894573  rs13303010  pancreatic carcinoma    803468  \n",
       "3   894573  rs13303010  pancreatic carcinoma    773641  \n",
       "4   894573  rs13303010  pancreatic carcinoma    761007  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "closest_endo = pd.read_table(closest_endo_f, sep=\"\\t\", header=None)\n",
    "closest_endo.columns = [\"chr\", \"start\", \"end\", \"transcript_id\", \"score\", \"strand\", \"snp_chr\", \"snp_start\", \"snp_end\",\n",
    "                        \"snp_id\", \"disease\", \"distance\"]\n",
    "closest_endo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>snp_chr</th>\n",
       "      <th>snp_start</th>\n",
       "      <th>snp_end</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>ENST00000456328.2_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>780396</td>\n",
       "      <td>780397</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>765988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12010</td>\n",
       "      <td>13670</td>\n",
       "      <td>ENST00000450305.2_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>780396</td>\n",
       "      <td>780397</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>766727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>ENST00000488147.1_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>780396</td>\n",
       "      <td>780397</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>750827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>31097</td>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>780396</td>\n",
       "      <td>780397</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>749300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30267</td>\n",
       "      <td>31109</td>\n",
       "      <td>ENST00000469289.1_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>780396</td>\n",
       "      <td>780397</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>749288</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  start    end        transcript_id snp_chr  snp_start  snp_end  \\\n",
       "0  chr1  11869  14409  ENST00000456328.2_1    chr1     780396   780397   \n",
       "1  chr1  12010  13670  ENST00000450305.2_1    chr1     780396   780397   \n",
       "2  chr1  14404  29570  ENST00000488147.1_1    chr1     780396   780397   \n",
       "3  chr1  29554  31097  ENST00000473358.1_1    chr1     780396   780397   \n",
       "4  chr1  30267  31109  ENST00000469289.1_1    chr1     780396   780397   \n",
       "\n",
       "        snp_id           disease  distance  \n",
       "0  rs141175086  circadian rhythm    765988  \n",
       "1  rs141175086  circadian rhythm    766727  \n",
       "2  rs141175086  circadian rhythm    750827  \n",
       "3  rs141175086  circadian rhythm    749300  \n",
       "4  rs141175086  circadian rhythm    749288  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "closest_gwas_gb = pd.read_table(closest_gwas_gb_f, sep=\"\\t\", header=None)\n",
    "closest_gwas_gb.columns = [\"chr\", \"start\", \"end\", \"transcript_id\", \"snp_chr\", \"snp_start\", \"snp_end\",\n",
    "                        \"snp_id\", \"disease\", \"distance\"]\n",
    "closest_gwas_gb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>snp_chr</th>\n",
       "      <th>snp_start</th>\n",
       "      <th>snp_end</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>ENST00000456328.2_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>880164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12010</td>\n",
       "      <td>13670</td>\n",
       "      <td>ENST00000450305.2_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>880903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>ENST00000488147.1_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>31097</td>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>863476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30267</td>\n",
       "      <td>31109</td>\n",
       "      <td>ENST00000469289.1_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>863464</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  start    end        transcript_id snp_chr  snp_start  snp_end  \\\n",
       "0  chr1  11869  14409  ENST00000456328.2_1    chr1     894572   894573   \n",
       "1  chr1  12010  13670  ENST00000450305.2_1    chr1     894572   894573   \n",
       "2  chr1  14404  29570  ENST00000488147.1_1    chr1     894572   894573   \n",
       "3  chr1  29554  31097  ENST00000473358.1_1    chr1     894572   894573   \n",
       "4  chr1  30267  31109  ENST00000469289.1_1    chr1     894572   894573   \n",
       "\n",
       "       snp_id               disease  distance  \n",
       "0  rs13303010  pancreatic carcinoma    880164  \n",
       "1  rs13303010  pancreatic carcinoma    880903  \n",
       "2  rs13303010  pancreatic carcinoma    865003  \n",
       "3  rs13303010  pancreatic carcinoma    863476  \n",
       "4  rs13303010  pancreatic carcinoma    863464  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "closest_cancer_gb = pd.read_table(closest_cancer_gb_f, sep=\"\\t\", header=None)\n",
    "closest_cancer_gb.columns = [\"chr\", \"start\", \"end\", \"transcript_id\", \"snp_chr\", \"snp_start\", \"snp_end\",\n",
    "                        \"snp_id\", \"disease\", \"distance\"]\n",
    "closest_cancer_gb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>snp_chr</th>\n",
       "      <th>snp_start</th>\n",
       "      <th>snp_end</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>ENST00000456328.2_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>880164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12010</td>\n",
       "      <td>13670</td>\n",
       "      <td>ENST00000450305.2_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>880903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>ENST00000488147.1_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>31097</td>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>863476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30267</td>\n",
       "      <td>31109</td>\n",
       "      <td>ENST00000469289.1_1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894572</td>\n",
       "      <td>894573</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>863464</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  start    end        transcript_id snp_chr  snp_start  snp_end  \\\n",
       "0  chr1  11869  14409  ENST00000456328.2_1    chr1     894572   894573   \n",
       "1  chr1  12010  13670  ENST00000450305.2_1    chr1     894572   894573   \n",
       "2  chr1  14404  29570  ENST00000488147.1_1    chr1     894572   894573   \n",
       "3  chr1  29554  31097  ENST00000473358.1_1    chr1     894572   894573   \n",
       "4  chr1  30267  31109  ENST00000469289.1_1    chr1     894572   894573   \n",
       "\n",
       "       snp_id               disease  distance  \n",
       "0  rs13303010  pancreatic carcinoma    880164  \n",
       "1  rs13303010  pancreatic carcinoma    880903  \n",
       "2  rs13303010  pancreatic carcinoma    865003  \n",
       "3  rs13303010  pancreatic carcinoma    863476  \n",
       "4  rs13303010  pancreatic carcinoma    863464  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "closest_endo_gb = pd.read_table(closest_endo_gb_f, sep=\"\\t\", header=None)\n",
    "closest_endo_gb.columns = [\"chr\", \"start\", \"end\", \"transcript_id\", \"snp_chr\", \"snp_start\", \"snp_end\",\n",
    "                        \"snp_id\", \"disease\", \"distance\"]\n",
    "closest_endo_gb.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. join metadata files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "closest_gwas = closest_gwas[[\"transcript_id\", \"snp_id\", \"disease\", \"distance\"]].drop_duplicates()\n",
    "closest_cancer = closest_cancer[[\"transcript_id\", \"snp_id\", \"disease\", \"distance\"]].drop_duplicates()\n",
    "closest_endo = closest_endo[[\"transcript_id\", \"snp_id\", \"disease\", \"distance\"]].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14730\n",
      "13808\n",
      "13348\n"
     ]
    }
   ],
   "source": [
    "print(len(closest_gwas))\n",
    "print(len(closest_cancer))\n",
    "print(len(closest_endo))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16770"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta = closest_gwas.merge(closest_cancer, on=\"transcript_id\", how=\"left\").merge(closest_endo, on=\"transcript_id\", how=\"left\")\n",
    "len(meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>closest_all_snp</th>\n",
       "      <th>closest_all_snp_disease</th>\n",
       "      <th>closest_all_snp_dist</th>\n",
       "      <th>closest_cancer_snp</th>\n",
       "      <th>closest_cancer_snp_disease</th>\n",
       "      <th>closest_cancer_snp_dist</th>\n",
       "      <th>closest_endo_snp</th>\n",
       "      <th>closest_endo_snp_disease</th>\n",
       "      <th>closest_endo_snp_dist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>750842</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865018</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000461467.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>744324</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000495576.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>689292</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000466430.5_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>659465</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000453576.2_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>646831</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         transcript_id closest_all_snp closest_all_snp_disease  \\\n",
       "0  ENST00000473358.1_1     rs141175086        circadian rhythm   \n",
       "1  ENST00000461467.1_1     rs141175086        circadian rhythm   \n",
       "2  ENST00000495576.1_1     rs141175086        circadian rhythm   \n",
       "3  ENST00000466430.5_1     rs141175086        circadian rhythm   \n",
       "4  ENST00000453576.2_1     rs141175086        circadian rhythm   \n",
       "\n",
       "   closest_all_snp_dist closest_cancer_snp closest_cancer_snp_disease  \\\n",
       "0                750842         rs13303010       pancreatic carcinoma   \n",
       "1                744324         rs13303010       pancreatic carcinoma   \n",
       "2                689292         rs13303010       pancreatic carcinoma   \n",
       "3                659465         rs13303010       pancreatic carcinoma   \n",
       "4                646831         rs13303010       pancreatic carcinoma   \n",
       "\n",
       "   closest_cancer_snp_dist closest_endo_snp closest_endo_snp_disease  \\\n",
       "0                   865018       rs13303010     pancreatic carcinoma   \n",
       "1                   858500       rs13303010     pancreatic carcinoma   \n",
       "2                   803468       rs13303010     pancreatic carcinoma   \n",
       "3                   773641       rs13303010     pancreatic carcinoma   \n",
       "4                   761007       rs13303010     pancreatic carcinoma   \n",
       "\n",
       "   closest_endo_snp_dist  \n",
       "0                 865018  \n",
       "1                 858500  \n",
       "2                 803468  \n",
       "3                 773641  \n",
       "4                 761007  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.columns = [\"transcript_id\", \"closest_all_snp\", \"closest_all_snp_disease\", \"closest_all_snp_dist\",\n",
    "                \"closest_cancer_snp\", \"closest_cancer_snp_disease\", \"closest_cancer_snp_dist\", \n",
    "                \"closest_endo_snp\", \"closest_endo_snp_disease\", \"closest_endo_snp_dist\"]\n",
    "meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12946\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>closest_all_snp</th>\n",
       "      <th>closest_all_snp_disease</th>\n",
       "      <th>closest_all_snp_dist</th>\n",
       "      <th>closest_cancer_snp</th>\n",
       "      <th>closest_cancer_snp_disease</th>\n",
       "      <th>closest_cancer_snp_dist</th>\n",
       "      <th>closest_endo_snp</th>\n",
       "      <th>closest_endo_snp_disease</th>\n",
       "      <th>closest_endo_snp_dist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>750842</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865018</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000461467.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>744324</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000495576.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>689292</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000466430.5_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>659465</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000453576.2_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>646831</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         transcript_id closest_all_snp closest_all_snp_disease  \\\n",
       "0  ENST00000473358.1_1     rs141175086        circadian rhythm   \n",
       "1  ENST00000461467.1_1     rs141175086        circadian rhythm   \n",
       "2  ENST00000495576.1_1     rs141175086        circadian rhythm   \n",
       "3  ENST00000466430.5_1     rs141175086        circadian rhythm   \n",
       "4  ENST00000453576.2_1     rs141175086        circadian rhythm   \n",
       "\n",
       "   closest_all_snp_dist closest_cancer_snp closest_cancer_snp_disease  \\\n",
       "0                750842         rs13303010       pancreatic carcinoma   \n",
       "1                744324         rs13303010       pancreatic carcinoma   \n",
       "2                689292         rs13303010       pancreatic carcinoma   \n",
       "3                659465         rs13303010       pancreatic carcinoma   \n",
       "4                646831         rs13303010       pancreatic carcinoma   \n",
       "\n",
       "   closest_cancer_snp_dist closest_endo_snp closest_endo_snp_disease  \\\n",
       "0                   865018       rs13303010     pancreatic carcinoma   \n",
       "1                   858500       rs13303010     pancreatic carcinoma   \n",
       "2                   803468       rs13303010     pancreatic carcinoma   \n",
       "3                   773641       rs13303010     pancreatic carcinoma   \n",
       "4                   761007       rs13303010     pancreatic carcinoma   \n",
       "\n",
       "   closest_endo_snp_dist  \n",
       "0                 865018  \n",
       "1                 858500  \n",
       "2                 803468  \n",
       "3                 773641  \n",
       "4                 761007  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# some transcripts have duplicate closest snps when there are ties, so drop these (keep first arbitrarily)\n",
    "meta.drop_duplicates(subset=\"transcript_id\", inplace=True)\n",
    "print(len(meta))\n",
    "meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "closest_gwas_gb = closest_gwas_gb[[\"transcript_id\", \"snp_id\", \"disease\", \"distance\"]].drop_duplicates()\n",
    "closest_cancer_gb = closest_cancer_gb[[\"transcript_id\", \"snp_id\", \"disease\", \"distance\"]].drop_duplicates()\n",
    "closest_endo_gb = closest_endo_gb[[\"transcript_id\", \"snp_id\", \"disease\", \"distance\"]].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "457486\n",
      "218313\n",
      "208200\n"
     ]
    }
   ],
   "source": [
    "print(len(closest_gwas_gb))\n",
    "print(len(closest_cancer_gb))\n",
    "print(len(closest_endo_gb))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "46026"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta = meta.merge(closest_gwas_gb, on=\"transcript_id\", how=\"left\").merge(closest_cancer_gb, on=\"transcript_id\", how=\"left\").merge(closest_endo_gb, on=\"transcript_id\", how=\"left\")\n",
    "len(meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>tss_closest_all_snp</th>\n",
       "      <th>tss_closest_all_snp_disease</th>\n",
       "      <th>tss_closest_all_snp_dist</th>\n",
       "      <th>tss_closest_cancer_snp</th>\n",
       "      <th>tss_closest_cancer_snp_disease</th>\n",
       "      <th>tss_closest_cancer_snp_dist</th>\n",
       "      <th>tss_closest_endo_snp</th>\n",
       "      <th>tss_closest_endo_snp_disease</th>\n",
       "      <th>tss_closest_endo_snp_dist</th>\n",
       "      <th>gb_closest_all_snp</th>\n",
       "      <th>gb_closest_all_snp_disease</th>\n",
       "      <th>gb_closest_all_snp_dist</th>\n",
       "      <th>gb_closest_cancer_snp</th>\n",
       "      <th>gb_closest_cancer_snp_disease</th>\n",
       "      <th>gb_closest_cancer_snp_dist</th>\n",
       "      <th>gb_closest_endo_snp</th>\n",
       "      <th>gb_closest_endo_snp_disease</th>\n",
       "      <th>gb_closest_endo_snp_dist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>750842</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865018</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865018</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>749300</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>863476</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>863476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000461467.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>744324</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>744324</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000495576.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>689292</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>689292</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000466430.5_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>659465</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>659465</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000453576.2_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>646831</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>646831</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         transcript_id tss_closest_all_snp tss_closest_all_snp_disease  \\\n",
       "0  ENST00000473358.1_1         rs141175086            circadian rhythm   \n",
       "1  ENST00000461467.1_1         rs141175086            circadian rhythm   \n",
       "2  ENST00000495576.1_1         rs141175086            circadian rhythm   \n",
       "3  ENST00000466430.5_1         rs141175086            circadian rhythm   \n",
       "4  ENST00000453576.2_1         rs141175086            circadian rhythm   \n",
       "\n",
       "   tss_closest_all_snp_dist tss_closest_cancer_snp  \\\n",
       "0                    750842             rs13303010   \n",
       "1                    744324             rs13303010   \n",
       "2                    689292             rs13303010   \n",
       "3                    659465             rs13303010   \n",
       "4                    646831             rs13303010   \n",
       "\n",
       "  tss_closest_cancer_snp_disease  tss_closest_cancer_snp_dist  \\\n",
       "0           pancreatic carcinoma                       865018   \n",
       "1           pancreatic carcinoma                       858500   \n",
       "2           pancreatic carcinoma                       803468   \n",
       "3           pancreatic carcinoma                       773641   \n",
       "4           pancreatic carcinoma                       761007   \n",
       "\n",
       "  tss_closest_endo_snp tss_closest_endo_snp_disease  \\\n",
       "0           rs13303010         pancreatic carcinoma   \n",
       "1           rs13303010         pancreatic carcinoma   \n",
       "2           rs13303010         pancreatic carcinoma   \n",
       "3           rs13303010         pancreatic carcinoma   \n",
       "4           rs13303010         pancreatic carcinoma   \n",
       "\n",
       "   tss_closest_endo_snp_dist gb_closest_all_snp gb_closest_all_snp_disease  \\\n",
       "0                     865018        rs141175086           circadian rhythm   \n",
       "1                     858500        rs141175086           circadian rhythm   \n",
       "2                     803468        rs141175086           circadian rhythm   \n",
       "3                     773641        rs141175086           circadian rhythm   \n",
       "4                     761007        rs141175086           circadian rhythm   \n",
       "\n",
       "   gb_closest_all_snp_dist gb_closest_cancer_snp  \\\n",
       "0                   749300            rs13303010   \n",
       "1                   744324            rs13303010   \n",
       "2                   689292            rs13303010   \n",
       "3                   659465            rs13303010   \n",
       "4                   646831            rs13303010   \n",
       "\n",
       "  gb_closest_cancer_snp_disease  gb_closest_cancer_snp_dist  \\\n",
       "0          pancreatic carcinoma                      863476   \n",
       "1          pancreatic carcinoma                      858500   \n",
       "2          pancreatic carcinoma                      803468   \n",
       "3          pancreatic carcinoma                      773641   \n",
       "4          pancreatic carcinoma                      761007   \n",
       "\n",
       "  gb_closest_endo_snp gb_closest_endo_snp_disease  gb_closest_endo_snp_dist  \n",
       "0          rs13303010        pancreatic carcinoma                    863476  \n",
       "1          rs13303010        pancreatic carcinoma                    858500  \n",
       "2          rs13303010        pancreatic carcinoma                    803468  \n",
       "3          rs13303010        pancreatic carcinoma                    773641  \n",
       "4          rs13303010        pancreatic carcinoma                    761007  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.columns = [\"transcript_id\", \"tss_closest_all_snp\", \"tss_closest_all_snp_disease\", \"tss_closest_all_snp_dist\",\n",
    "                \"tss_closest_cancer_snp\", \"tss_closest_cancer_snp_disease\", \"tss_closest_cancer_snp_dist\", \n",
    "                \"tss_closest_endo_snp\", \"tss_closest_endo_snp_disease\", \"tss_closest_endo_snp_dist\",\n",
    "                \"gb_closest_all_snp\", \"gb_closest_all_snp_disease\", \"gb_closest_all_snp_dist\",\n",
    "                \"gb_closest_cancer_snp\", \"gb_closest_cancer_snp_disease\", \"gb_closest_cancer_snp_dist\",\n",
    "                \"gb_closest_endo_snp\", \"gb_closest_endo_snp_disease\", \"gb_closest_endo_snp_dist\"]\n",
    "meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12946\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>tss_closest_all_snp</th>\n",
       "      <th>tss_closest_all_snp_disease</th>\n",
       "      <th>tss_closest_all_snp_dist</th>\n",
       "      <th>tss_closest_cancer_snp</th>\n",
       "      <th>tss_closest_cancer_snp_disease</th>\n",
       "      <th>tss_closest_cancer_snp_dist</th>\n",
       "      <th>tss_closest_endo_snp</th>\n",
       "      <th>tss_closest_endo_snp_disease</th>\n",
       "      <th>tss_closest_endo_snp_dist</th>\n",
       "      <th>gb_closest_all_snp</th>\n",
       "      <th>gb_closest_all_snp_disease</th>\n",
       "      <th>gb_closest_all_snp_dist</th>\n",
       "      <th>gb_closest_cancer_snp</th>\n",
       "      <th>gb_closest_cancer_snp_disease</th>\n",
       "      <th>gb_closest_cancer_snp_dist</th>\n",
       "      <th>gb_closest_endo_snp</th>\n",
       "      <th>gb_closest_endo_snp_disease</th>\n",
       "      <th>gb_closest_endo_snp_dist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000473358.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>750842</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865018</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>865018</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>749300</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>863476</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>863476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000461467.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>744324</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>744324</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>858500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000495576.1_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>689292</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>689292</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>803468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000466430.5_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>659465</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>659465</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>773641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000453576.2_1</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>646831</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "      <td>rs141175086</td>\n",
       "      <td>circadian rhythm</td>\n",
       "      <td>646831</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "      <td>rs13303010</td>\n",
       "      <td>pancreatic carcinoma</td>\n",
       "      <td>761007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         transcript_id tss_closest_all_snp tss_closest_all_snp_disease  \\\n",
       "0  ENST00000473358.1_1         rs141175086            circadian rhythm   \n",
       "1  ENST00000461467.1_1         rs141175086            circadian rhythm   \n",
       "2  ENST00000495576.1_1         rs141175086            circadian rhythm   \n",
       "3  ENST00000466430.5_1         rs141175086            circadian rhythm   \n",
       "4  ENST00000453576.2_1         rs141175086            circadian rhythm   \n",
       "\n",
       "   tss_closest_all_snp_dist tss_closest_cancer_snp  \\\n",
       "0                    750842             rs13303010   \n",
       "1                    744324             rs13303010   \n",
       "2                    689292             rs13303010   \n",
       "3                    659465             rs13303010   \n",
       "4                    646831             rs13303010   \n",
       "\n",
       "  tss_closest_cancer_snp_disease  tss_closest_cancer_snp_dist  \\\n",
       "0           pancreatic carcinoma                       865018   \n",
       "1           pancreatic carcinoma                       858500   \n",
       "2           pancreatic carcinoma                       803468   \n",
       "3           pancreatic carcinoma                       773641   \n",
       "4           pancreatic carcinoma                       761007   \n",
       "\n",
       "  tss_closest_endo_snp tss_closest_endo_snp_disease  \\\n",
       "0           rs13303010         pancreatic carcinoma   \n",
       "1           rs13303010         pancreatic carcinoma   \n",
       "2           rs13303010         pancreatic carcinoma   \n",
       "3           rs13303010         pancreatic carcinoma   \n",
       "4           rs13303010         pancreatic carcinoma   \n",
       "\n",
       "   tss_closest_endo_snp_dist gb_closest_all_snp gb_closest_all_snp_disease  \\\n",
       "0                     865018        rs141175086           circadian rhythm   \n",
       "1                     858500        rs141175086           circadian rhythm   \n",
       "2                     803468        rs141175086           circadian rhythm   \n",
       "3                     773641        rs141175086           circadian rhythm   \n",
       "4                     761007        rs141175086           circadian rhythm   \n",
       "\n",
       "   gb_closest_all_snp_dist gb_closest_cancer_snp  \\\n",
       "0                   749300            rs13303010   \n",
       "1                   744324            rs13303010   \n",
       "2                   689292            rs13303010   \n",
       "3                   659465            rs13303010   \n",
       "4                   646831            rs13303010   \n",
       "\n",
       "  gb_closest_cancer_snp_disease  gb_closest_cancer_snp_dist  \\\n",
       "0          pancreatic carcinoma                      863476   \n",
       "1          pancreatic carcinoma                      858500   \n",
       "2          pancreatic carcinoma                      803468   \n",
       "3          pancreatic carcinoma                      773641   \n",
       "4          pancreatic carcinoma                      761007   \n",
       "\n",
       "  gb_closest_endo_snp gb_closest_endo_snp_disease  gb_closest_endo_snp_dist  \n",
       "0          rs13303010        pancreatic carcinoma                    863476  \n",
       "1          rs13303010        pancreatic carcinoma                    858500  \n",
       "2          rs13303010        pancreatic carcinoma                    803468  \n",
       "3          rs13303010        pancreatic carcinoma                    773641  \n",
       "4          rs13303010        pancreatic carcinoma                    761007  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# some transcripts have duplicate closest snps when there are ties, so drop these (keep first arbitrarily)\n",
    "meta.drop_duplicates(subset=\"transcript_id\", inplace=True)\n",
    "print(len(meta))\n",
    "meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>tss_closest_all_snp</th>\n",
       "      <th>tss_closest_all_snp_disease</th>\n",
       "      <th>tss_closest_all_snp_dist</th>\n",
       "      <th>tss_closest_cancer_snp</th>\n",
       "      <th>tss_closest_cancer_snp_disease</th>\n",
       "      <th>tss_closest_cancer_snp_dist</th>\n",
       "      <th>tss_closest_endo_snp</th>\n",
       "      <th>tss_closest_endo_snp_disease</th>\n",
       "      <th>tss_closest_endo_snp_dist</th>\n",
       "      <th>gb_closest_all_snp</th>\n",
       "      <th>gb_closest_all_snp_disease</th>\n",
       "      <th>gb_closest_all_snp_dist</th>\n",
       "      <th>gb_closest_cancer_snp</th>\n",
       "      <th>gb_closest_cancer_snp_disease</th>\n",
       "      <th>gb_closest_cancer_snp_dist</th>\n",
       "      <th>gb_closest_endo_snp</th>\n",
       "      <th>gb_closest_endo_snp_disease</th>\n",
       "      <th>gb_closest_endo_snp_dist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6682</th>\n",
       "      <td>DIGIT</td>\n",
       "      <td>rs61977111</td>\n",
       "      <td>balding measurement</td>\n",
       "      <td>236</td>\n",
       "      <td>rs34877263</td>\n",
       "      <td>breast carcinoma, mortality</td>\n",
       "      <td>253846</td>\n",
       "      <td>rs11622887</td>\n",
       "      <td>response to irinotecan, non-small cell lung ca...</td>\n",
       "      <td>705468</td>\n",
       "      <td>rs61977111</td>\n",
       "      <td>balding measurement</td>\n",
       "      <td>0</td>\n",
       "      <td>rs34877263</td>\n",
       "      <td>breast carcinoma, mortality</td>\n",
       "      <td>250881</td>\n",
       "      <td>rs11622887</td>\n",
       "      <td>response to irinotecan, non-small cell lung ca...</td>\n",
       "      <td>702503</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     transcript_id tss_closest_all_snp tss_closest_all_snp_disease  \\\n",
       "6682         DIGIT          rs61977111         balding measurement   \n",
       "\n",
       "      tss_closest_all_snp_dist tss_closest_cancer_snp  \\\n",
       "6682                       236             rs34877263   \n",
       "\n",
       "     tss_closest_cancer_snp_disease  tss_closest_cancer_snp_dist  \\\n",
       "6682    breast carcinoma, mortality                       253846   \n",
       "\n",
       "     tss_closest_endo_snp                       tss_closest_endo_snp_disease  \\\n",
       "6682           rs11622887  response to irinotecan, non-small cell lung ca...   \n",
       "\n",
       "      tss_closest_endo_snp_dist gb_closest_all_snp gb_closest_all_snp_disease  \\\n",
       "6682                     705468         rs61977111        balding measurement   \n",
       "\n",
       "      gb_closest_all_snp_dist gb_closest_cancer_snp  \\\n",
       "6682                        0            rs34877263   \n",
       "\n",
       "     gb_closest_cancer_snp_disease  gb_closest_cancer_snp_dist  \\\n",
       "6682   breast carcinoma, mortality                      250881   \n",
       "\n",
       "     gb_closest_endo_snp                        gb_closest_endo_snp_disease  \\\n",
       "6682          rs11622887  response to irinotecan, non-small cell lung ca...   \n",
       "\n",
       "      gb_closest_endo_snp_dist  \n",
       "6682                    702503  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta[meta[\"transcript_id\"] == \"DIGIT\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. merge metadata w/ hit info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "no hit           11196\n",
       "lenient hit       1428\n",
       "stringent hit       73\n",
       "Name: is_hit, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.is_hit.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12697\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group_id</th>\n",
       "      <th>ctrl_status</th>\n",
       "      <th>endo_ctrl_val</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>cleaner_transcript_biotype</th>\n",
       "      <th>cleaner_gene_biotype</th>\n",
       "      <th>BFP+_score__rep1</th>\n",
       "      <th>...</th>\n",
       "      <th>tss_closest_endo_snp_dist</th>\n",
       "      <th>gb_closest_all_snp</th>\n",
       "      <th>gb_closest_all_snp_disease</th>\n",
       "      <th>gb_closest_all_snp_dist</th>\n",
       "      <th>gb_closest_cancer_snp</th>\n",
       "      <th>gb_closest_cancer_snp_disease</th>\n",
       "      <th>gb_closest_cancer_snp_dist</th>\n",
       "      <th>gb_closest_endo_snp</th>\n",
       "      <th>gb_closest_endo_snp_disease</th>\n",
       "      <th>gb_closest_endo_snp_dist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000008.11:+:54457934</td>\n",
       "      <td>control</td>\n",
       "      <td>True</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>ENSG00000164736.5_1</td>\n",
       "      <td>SOX17-001</td>\n",
       "      <td>ENST00000297316.4_1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>232.279434</td>\n",
       "      <td>...</td>\n",
       "      <td>125338</td>\n",
       "      <td>rs10958409</td>\n",
       "      <td>brain aneurysm</td>\n",
       "      <td>43405</td>\n",
       "      <td>rs28571765</td>\n",
       "      <td>multiple myeloma</td>\n",
       "      <td>75833</td>\n",
       "      <td>rs188586526</td>\n",
       "      <td>lung carcinoma</td>\n",
       "      <td>122385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000003.12:-:27722707</td>\n",
       "      <td>control</td>\n",
       "      <td>True</td>\n",
       "      <td>EOMES</td>\n",
       "      <td>ENSG00000163508.12_1</td>\n",
       "      <td>EOMES-004</td>\n",
       "      <td>ENST00000461503.2_1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>156.570722</td>\n",
       "      <td>...</td>\n",
       "      <td>436985</td>\n",
       "      <td>rs3806624</td>\n",
       "      <td>Hodgkins lymphoma</td>\n",
       "      <td>421</td>\n",
       "      <td>rs3806624</td>\n",
       "      <td>Hodgkins lymphoma</td>\n",
       "      <td>421</td>\n",
       "      <td>rs481519</td>\n",
       "      <td>lung carcinoma, estrogen-receptor negative bre...</td>\n",
       "      <td>430787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000003.12:-:27722322</td>\n",
       "      <td>control</td>\n",
       "      <td>True</td>\n",
       "      <td>EOMES</td>\n",
       "      <td>ENSG00000163508.12_1</td>\n",
       "      <td>EOMES-001</td>\n",
       "      <td>ENST00000295743.8_1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>43.594914</td>\n",
       "      <td>...</td>\n",
       "      <td>436600</td>\n",
       "      <td>rs2371108</td>\n",
       "      <td>leukocyte count</td>\n",
       "      <td>423</td>\n",
       "      <td>rs3806624</td>\n",
       "      <td>Hodgkins lymphoma</td>\n",
       "      <td>634</td>\n",
       "      <td>rs481519</td>\n",
       "      <td>lung carcinoma, estrogen-receptor negative bre...</td>\n",
       "      <td>430227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000016.10:+:23452748</td>\n",
       "      <td>experimental</td>\n",
       "      <td>False</td>\n",
       "      <td>CTD-2270L9.4</td>\n",
       "      <td>ENSG00000260136.5_1</td>\n",
       "      <td>CTD-2270L9.4-001</td>\n",
       "      <td>ENST00000570080.1_1</td>\n",
       "      <td>promoter_overlap</td>\n",
       "      <td>promoter_overlap</td>\n",
       "      <td>3.702626</td>\n",
       "      <td>...</td>\n",
       "      <td>3075192</td>\n",
       "      <td>rs250585</td>\n",
       "      <td>egg allergy measurement</td>\n",
       "      <td>63004</td>\n",
       "      <td>rs35557655</td>\n",
       "      <td>estrogen-receptor negative breast cancer, mort...</td>\n",
       "      <td>656660</td>\n",
       "      <td>rs35874699</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>3070353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000016.10:+:23452748</td>\n",
       "      <td>experimental</td>\n",
       "      <td>False</td>\n",
       "      <td>CTD-2270L9.4</td>\n",
       "      <td>ENSG00000260136.5_1</td>\n",
       "      <td>CTD-2270L9.4-002</td>\n",
       "      <td>ENST00000562842.1_1</td>\n",
       "      <td>promoter_overlap</td>\n",
       "      <td>promoter_overlap</td>\n",
       "      <td>3.702626</td>\n",
       "      <td>...</td>\n",
       "      <td>3075192</td>\n",
       "      <td>rs250585</td>\n",
       "      <td>egg allergy measurement</td>\n",
       "      <td>63253</td>\n",
       "      <td>rs35557655</td>\n",
       "      <td>estrogen-receptor negative breast cancer, mort...</td>\n",
       "      <td>656909</td>\n",
       "      <td>rs35874699</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>3070335</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 45 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  group_id   ctrl_status  endo_ctrl_val     gene_name  \\\n",
       "0  NC_000008.11:+:54457934       control           True         SOX17   \n",
       "1  NC_000003.12:-:27722707       control           True         EOMES   \n",
       "2  NC_000003.12:-:27722322       control           True         EOMES   \n",
       "3  NC_000016.10:+:23452748  experimental          False  CTD-2270L9.4   \n",
       "4  NC_000016.10:+:23452748  experimental          False  CTD-2270L9.4   \n",
       "\n",
       "                gene_id   transcript_name        transcript_id  \\\n",
       "0   ENSG00000164736.5_1         SOX17-001  ENST00000297316.4_1   \n",
       "1  ENSG00000163508.12_1         EOMES-004  ENST00000461503.2_1   \n",
       "2  ENSG00000163508.12_1         EOMES-001  ENST00000295743.8_1   \n",
       "3   ENSG00000260136.5_1  CTD-2270L9.4-001  ENST00000570080.1_1   \n",
       "4   ENSG00000260136.5_1  CTD-2270L9.4-002  ENST00000562842.1_1   \n",
       "\n",
       "  cleaner_transcript_biotype cleaner_gene_biotype  BFP+_score__rep1  \\\n",
       "0             protein_coding       protein_coding        232.279434   \n",
       "1             protein_coding       protein_coding        156.570722   \n",
       "2             protein_coding       protein_coding         43.594914   \n",
       "3           promoter_overlap     promoter_overlap          3.702626   \n",
       "4           promoter_overlap     promoter_overlap          3.702626   \n",
       "\n",
       "             ...             tss_closest_endo_snp_dist  gb_closest_all_snp  \\\n",
       "0            ...                                125338          rs10958409   \n",
       "1            ...                                436985           rs3806624   \n",
       "2            ...                                436600           rs2371108   \n",
       "3            ...                               3075192            rs250585   \n",
       "4            ...                               3075192            rs250585   \n",
       "\n",
       "   gb_closest_all_snp_disease  gb_closest_all_snp_dist  gb_closest_cancer_snp  \\\n",
       "0              brain aneurysm                    43405             rs28571765   \n",
       "1           Hodgkins lymphoma                      421              rs3806624   \n",
       "2             leukocyte count                      423              rs3806624   \n",
       "3     egg allergy measurement                    63004             rs35557655   \n",
       "4     egg allergy measurement                    63253             rs35557655   \n",
       "\n",
       "                       gb_closest_cancer_snp_disease  \\\n",
       "0                                   multiple myeloma   \n",
       "1                                  Hodgkins lymphoma   \n",
       "2                                  Hodgkins lymphoma   \n",
       "3  estrogen-receptor negative breast cancer, mort...   \n",
       "4  estrogen-receptor negative breast cancer, mort...   \n",
       "\n",
       "   gb_closest_cancer_snp_dist  gb_closest_endo_snp  \\\n",
       "0                       75833          rs188586526   \n",
       "1                         421             rs481519   \n",
       "2                         634             rs481519   \n",
       "3                      656660           rs35874699   \n",
       "4                      656909           rs35874699   \n",
       "\n",
       "                         gb_closest_endo_snp_disease  gb_closest_endo_snp_dist  \n",
       "0                                     lung carcinoma                    122385  \n",
       "1  lung carcinoma, estrogen-receptor negative bre...                    430787  \n",
       "2  lung carcinoma, estrogen-receptor negative bre...                    430227  \n",
       "3                                  colorectal cancer                   3070353  \n",
       "4                                  colorectal cancer                   3070335  \n",
       "\n",
       "[5 rows x 45 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = data.merge(meta, on=\"transcript_id\", how=\"left\")\n",
    "print(len(data))\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "no hit           11196\n",
       "lenient hit       1428\n",
       "stringent hit       73\n",
       "Name: is_hit, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.is_hit.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. plots to examine gwas closeness"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "order = [\"lncRNA_good_csf\", \"protein_coding\"]\n",
    "hue_order = [\"no hit\", \"stringent hit\"]\n",
    "pal = {\"no hit\": \"gray\", \"lenient hit\": sns.color_palette(\"Set2\")[2], \"stringent hit\": \"black\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "hue_order2 = [\"no hit\", \"hit\"]\n",
    "pal2 = {\"no hit\": \"gray\", \"hit\": \"black\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "def snp_within_x(row, col, x):\n",
    "    if row[col] < x:\n",
    "        return \"yes\"\n",
    "    else:\n",
    "        return \"no\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "data[\"gwas_within_5_gb\"] = data.apply(snp_within_x, col=\"gb_closest_all_snp_dist\", x=5000, axis=1)\n",
    "data[\"cancer_within_5_gb\"] = data.apply(snp_within_x, col=\"gb_closest_cancer_snp_dist\", x=5000, axis=1)\n",
    "data[\"endo_within_5_gb\"] = data.apply(snp_within_x, col=\"gb_closest_endo_snp_dist\", x=5000, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.00232898220262\n",
      "1.0\n"
     ]
    },
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       "  \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Created with matplotlib (http://matplotlib.org/) -->\n",
       "<svg height=\"157pt\" version=\"1.1\" viewBox=\"0 0 138 157\" width=\"138pt\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       " <defs>\n",
       "  <style type=\"text/css\">\n",
       "*{stroke-linecap:butt;stroke-linejoin:round;}\n",
       "  </style>\n",
       " </defs>\n",
       " <g id=\"figure_1\">\n",
       "  <g id=\"patch_1\">\n",
       "   <path d=\"M -0 157.981035 \n",
       "L 138.403594 157.981035 \n",
       "L 138.403594 0 \n",
       "L -0 0 \n",
       "z\n",
       "\" style=\"fill:#ffffff;\"/>\n",
       "  </g>\n",
       "  <g id=\"axes_1\">\n",
       "   <g id=\"patch_2\">\n",
       "    <path d=\"M 30.353594 118.8585 \n",
       "L 74.739957 118.8585 \n",
       "L 74.739957 18.2925 \n",
       "L 30.353594 18.2925 \n",
       "z\n",
       "\" style=\"fill:#ffffff;\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_1\">\n",
       "    <g id=\"xtick_1\">\n",
       "     <g id=\"line2d_1\">\n",
       "      <defs>\n",
       "       <path d=\"M 0 0 \n",
       "L 0 3 \n",
       "\" id=\"m21aee0c6b8\" style=\"stroke:#262626;\"/>\n",
       "      </defs>\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"41.338098\" xlink:href=\"#m21aee0c6b8\" y=\"118.8585\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_1\">\n",
       "      <!-- no hit -->\n",
       "      <defs>\n",
       "       <path d=\"M 6.453125 52.296875 \n",
       "L 14.796875 52.296875 \n",
       "L 14.796875 44.875 \n",
       "Q 18.5 49.46875 22.65625 51.46875 \n",
       "Q 26.8125 53.46875 31.890625 53.46875 \n",
       "Q 43.015625 53.46875 46.921875 45.703125 \n",
       "Q 49.078125 41.453125 49.078125 33.546875 \n",
       "L 49.078125 0 \n",
       "L 40.140625 0 \n",
       "L 40.140625 32.953125 \n",
       "Q 40.140625 37.75 38.71875 40.671875 \n",
       "Q 36.375 45.5625 30.21875 45.5625 \n",
       "Q 27.09375 45.5625 25.09375 44.921875 \n",
       "Q 21.484375 43.84375 18.75 40.625 \n",
       "Q 16.546875 38.03125 15.890625 35.265625 \n",
       "Q 15.234375 32.515625 15.234375 27.390625 \n",
       "L 15.234375 0 \n",
       "L 6.453125 0 \n",
       "z\n",
       "M 27.09375 53.5625 \n",
       "z\n",
       "\" id=\"Helvetica-6e\"/>\n",
       "       <path d=\"M 27.203125 5.671875 \n",
       "Q 35.9375 5.671875 39.1875 12.28125 \n",
       "Q 42.4375 18.890625 42.4375 27 \n",
       "Q 42.4375 34.328125 40.09375 38.921875 \n",
       "Q 36.375 46.140625 27.296875 46.140625 \n",
       "Q 19.234375 46.140625 15.578125 39.984375 \n",
       "Q 11.921875 33.84375 11.921875 25.140625 \n",
       "Q 11.921875 16.796875 15.578125 11.234375 \n",
       "Q 19.234375 5.671875 27.203125 5.671875 \n",
       "z\n",
       "M 27.546875 53.8125 \n",
       "Q 37.640625 53.8125 44.625 47.0625 \n",
       "Q 51.609375 40.328125 51.609375 27.25 \n",
       "Q 51.609375 14.59375 45.453125 6.34375 \n",
       "Q 39.3125 -1.90625 26.375 -1.90625 \n",
       "Q 15.578125 -1.90625 9.21875 5.390625 \n",
       "Q 2.875 12.703125 2.875 25 \n",
       "Q 2.875 38.1875 9.5625 46 \n",
       "Q 16.265625 53.8125 27.546875 53.8125 \n",
       "z\n",
       "M 27.25 53.5625 \n",
       "z\n",
       "\" id=\"Helvetica-6f\"/>\n",
       "       <path id=\"Helvetica-20\"/>\n",
       "       <path d=\"M 6.453125 71.96875 \n",
       "L 15.234375 71.96875 \n",
       "L 15.234375 45.21875 \n",
       "Q 18.359375 49.171875 20.84375 50.78125 \n",
       "Q 25.09375 53.5625 31.453125 53.5625 \n",
       "Q 42.828125 53.5625 46.875 45.609375 \n",
       "Q 49.078125 41.265625 49.078125 33.546875 \n",
       "L 49.078125 0 \n",
       "L 40.046875 0 \n",
       "L 40.046875 32.953125 \n",
       "Q 40.046875 38.71875 38.578125 41.40625 \n",
       "Q 36.1875 45.703125 29.59375 45.703125 \n",
       "Q 24.125 45.703125 19.671875 41.9375 \n",
       "Q 15.234375 38.1875 15.234375 27.734375 \n",
       "L 15.234375 0 \n",
       "L 6.453125 0 \n",
       "z\n",
       "\" id=\"Helvetica-68\"/>\n",
       "       <path d=\"M 6.453125 52.046875 \n",
       "L 15.375 52.046875 \n",
       "L 15.375 0 \n",
       "L 6.453125 0 \n",
       "z\n",
       "M 6.453125 71.734375 \n",
       "L 15.375 71.734375 \n",
       "L 15.375 61.765625 \n",
       "L 6.453125 61.765625 \n",
       "z\n",
       "\" id=\"Helvetica-69\"/>\n",
       "       <path d=\"M 8.203125 66.890625 \n",
       "L 17.09375 66.890625 \n",
       "L 17.09375 52.296875 \n",
       "L 25.4375 52.296875 \n",
       "L 25.4375 45.125 \n",
       "L 17.09375 45.125 \n",
       "L 17.09375 10.984375 \n",
       "Q 17.09375 8.25 18.953125 7.328125 \n",
       "Q 19.96875 6.78125 22.359375 6.78125 \n",
       "Q 23 6.78125 23.734375 6.8125 \n",
       "Q 24.46875 6.84375 25.4375 6.9375 \n",
       "L 25.4375 0 \n",
       "Q 23.921875 -0.4375 22.28125 -0.625 \n",
       "Q 20.65625 -0.828125 18.75 -0.828125 \n",
       "Q 12.59375 -0.828125 10.390625 2.3125 \n",
       "Q 8.203125 5.46875 8.203125 10.5 \n",
       "L 8.203125 45.125 \n",
       "L 1.125 45.125 \n",
       "L 1.125 52.296875 \n",
       "L 8.203125 52.296875 \n",
       "z\n",
       "\" id=\"Helvetica-74\"/>\n",
       "      </defs>\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(30.075876 138.279829)rotate(-45)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-6e\"/>\n",
       "       <use x=\"55.615234\" xlink:href=\"#Helvetica-6f\"/>\n",
       "       <use x=\"111.230469\" xlink:href=\"#Helvetica-20\"/>\n",
       "       <use x=\"139.013672\" xlink:href=\"#Helvetica-68\"/>\n",
       "       <use x=\"194.628906\" xlink:href=\"#Helvetica-69\"/>\n",
       "       <use x=\"216.845703\" xlink:href=\"#Helvetica-74\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_2\">\n",
       "     <g id=\"line2d_2\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"63.755453\" xlink:href=\"#m21aee0c6b8\" y=\"118.8585\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_2\">\n",
       "      <!-- stringent hit -->\n",
       "      <defs>\n",
       "       <path d=\"M 11.671875 16.40625 \n",
       "Q 12.0625 12.015625 13.875 9.671875 \n",
       "Q 17.1875 5.421875 25.390625 5.421875 \n",
       "Q 30.28125 5.421875 33.984375 7.546875 \n",
       "Q 37.703125 9.671875 37.703125 14.109375 \n",
       "Q 37.703125 17.484375 34.71875 19.234375 \n",
       "Q 32.8125 20.3125 27.203125 21.734375 \n",
       "L 20.21875 23.484375 \n",
       "Q 13.53125 25.140625 10.359375 27.203125 \n",
       "Q 4.6875 30.765625 4.6875 37.0625 \n",
       "Q 4.6875 44.484375 10.03125 49.0625 \n",
       "Q 15.375 53.65625 24.421875 53.65625 \n",
       "Q 36.234375 53.65625 41.453125 46.734375 \n",
       "Q 44.734375 42.328125 44.625 37.25 \n",
       "L 36.328125 37.25 \n",
       "Q 36.078125 40.234375 34.234375 42.671875 \n",
       "Q 31.203125 46.140625 23.734375 46.140625 \n",
       "Q 18.75 46.140625 16.1875 44.234375 \n",
       "Q 13.625 42.328125 13.625 39.203125 \n",
       "Q 13.625 35.796875 17 33.734375 \n",
       "Q 18.953125 32.515625 22.75 31.59375 \n",
       "L 28.5625 30.171875 \n",
       "Q 38.03125 27.875 41.265625 25.734375 \n",
       "Q 46.390625 22.359375 46.390625 15.140625 \n",
       "Q 46.390625 8.15625 41.09375 3.078125 \n",
       "Q 35.796875 -2 24.953125 -2 \n",
       "Q 13.28125 -2 8.421875 3.296875 \n",
       "Q 3.5625 8.59375 3.21875 16.40625 \n",
       "z\n",
       "M 24.65625 53.5625 \n",
       "z\n",
       "\" id=\"Helvetica-73\"/>\n",
       "       <path d=\"M 6.6875 52.296875 \n",
       "L 15.046875 52.296875 \n",
       "L 15.046875 43.265625 \n",
       "Q 16.0625 45.90625 20.0625 49.6875 \n",
       "Q 24.078125 53.46875 29.296875 53.46875 \n",
       "Q 29.546875 53.46875 30.125 53.421875 \n",
       "Q 30.71875 53.375 32.125 53.21875 \n",
       "L 32.125 43.953125 \n",
       "Q 31.34375 44.09375 30.6875 44.140625 \n",
       "Q 30.03125 44.1875 29.25 44.1875 \n",
       "Q 22.609375 44.1875 19.046875 39.90625 \n",
       "Q 15.484375 35.640625 15.484375 30.078125 \n",
       "L 15.484375 0 \n",
       "L 6.6875 0 \n",
       "z\n",
       "\" id=\"Helvetica-72\"/>\n",
       "       <path d=\"M 24.90625 53.21875 \n",
       "Q 31.0625 53.21875 35.640625 50.203125 \n",
       "Q 38.140625 48.484375 40.71875 45.21875 \n",
       "L 40.71875 51.8125 \n",
       "L 48.828125 51.8125 \n",
       "L 48.828125 4.25 \n",
       "Q 48.828125 -5.71875 45.90625 -11.46875 \n",
       "Q 40.4375 -22.125 25.25 -22.125 \n",
       "Q 16.796875 -22.125 11.03125 -18.328125 \n",
       "Q 5.28125 -14.546875 4.59375 -6.5 \n",
       "L 13.53125 -6.5 \n",
       "Q 14.15625 -10.015625 16.0625 -11.921875 \n",
       "Q 19.046875 -14.84375 25.4375 -14.84375 \n",
       "Q 35.546875 -14.84375 38.671875 -7.71875 \n",
       "Q 40.53125 -3.515625 40.375 7.28125 \n",
       "Q 37.75 3.265625 34.03125 1.3125 \n",
       "Q 30.328125 -0.640625 24.21875 -0.640625 \n",
       "Q 15.71875 -0.640625 9.34375 5.390625 \n",
       "Q 2.984375 11.421875 2.984375 25.34375 \n",
       "Q 2.984375 38.484375 9.40625 45.84375 \n",
       "Q 15.828125 53.21875 24.90625 53.21875 \n",
       "z\n",
       "M 40.71875 26.375 \n",
       "Q 40.71875 36.078125 36.71875 40.765625 \n",
       "Q 32.71875 45.453125 26.515625 45.453125 \n",
       "Q 17.234375 45.453125 13.8125 36.765625 \n",
       "Q 12.015625 32.125 12.015625 24.609375 \n",
       "Q 12.015625 15.765625 15.59375 11.15625 \n",
       "Q 19.1875 6.546875 25.25 6.546875 \n",
       "Q 34.71875 6.546875 38.578125 15.09375 \n",
       "Q 40.71875 19.921875 40.71875 26.375 \n",
       "z\n",
       "M 25.921875 53.5625 \n",
       "z\n",
       "\" id=\"Helvetica-67\"/>\n",
       "       <path d=\"M 28.21875 53.46875 \n",
       "Q 33.796875 53.46875 39.015625 50.859375 \n",
       "Q 44.234375 48.25 46.96875 44.09375 \n",
       "Q 49.609375 40.140625 50.484375 34.859375 \n",
       "Q 51.265625 31.25 51.265625 23.34375 \n",
       "L 12.9375 23.34375 \n",
       "Q 13.1875 15.375 16.703125 10.5625 \n",
       "Q 20.21875 5.765625 27.59375 5.765625 \n",
       "Q 34.46875 5.765625 38.578125 10.296875 \n",
       "Q 40.921875 12.9375 41.890625 16.40625 \n",
       "L 50.53125 16.40625 \n",
       "Q 50.203125 13.53125 48.265625 9.984375 \n",
       "Q 46.34375 6.453125 43.953125 4.203125 \n",
       "Q 39.9375 0.296875 34.03125 -1.078125 \n",
       "Q 30.859375 -1.859375 26.859375 -1.859375 \n",
       "Q 17.09375 -1.859375 10.296875 5.25 \n",
       "Q 3.515625 12.359375 3.515625 25.140625 \n",
       "Q 3.515625 37.75 10.34375 45.609375 \n",
       "Q 17.1875 53.46875 28.21875 53.46875 \n",
       "z\n",
       "M 42.234375 30.328125 \n",
       "Q 41.703125 36.03125 39.75 39.453125 \n",
       "Q 36.140625 45.796875 27.6875 45.796875 \n",
       "Q 21.625 45.796875 17.515625 41.421875 \n",
       "Q 13.421875 37.0625 13.1875 30.328125 \n",
       "z\n",
       "M 27.390625 53.5625 \n",
       "z\n",
       "\" id=\"Helvetica-65\"/>\n",
       "      </defs>\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(40.886294 149.84235)rotate(-45)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-73\"/>\n",
       "       <use x=\"50\" xlink:href=\"#Helvetica-74\"/>\n",
       "       <use x=\"77.783203\" xlink:href=\"#Helvetica-72\"/>\n",
       "       <use x=\"111.083984\" xlink:href=\"#Helvetica-69\"/>\n",
       "       <use x=\"133.300781\" xlink:href=\"#Helvetica-6e\"/>\n",
       "       <use x=\"188.916016\" xlink:href=\"#Helvetica-67\"/>\n",
       "       <use x=\"244.53125\" xlink:href=\"#Helvetica-65\"/>\n",
       "       <use x=\"300.146484\" xlink:href=\"#Helvetica-6e\"/>\n",
       "       <use x=\"355.761719\" xlink:href=\"#Helvetica-74\"/>\n",
       "       <use x=\"383.544922\" xlink:href=\"#Helvetica-20\"/>\n",
       "       <use x=\"411.328125\" xlink:href=\"#Helvetica-68\"/>\n",
       "       <use x=\"466.943359\" xlink:href=\"#Helvetica-69\"/>\n",
       "       <use x=\"489.160156\" xlink:href=\"#Helvetica-74\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_2\">\n",
       "    <g id=\"ytick_1\">\n",
       "     <g id=\"line2d_3\">\n",
       "      <defs>\n",
       "       <path d=\"M 0 0 \n",
       "L -3 0 \n",
       "\" id=\"m95115c8f30\" style=\"stroke:#262626;\"/>\n",
       "      </defs>\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"30.353594\" xlink:href=\"#m95115c8f30\" y=\"118.8585\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_3\">\n",
       "      <!-- 0 -->\n",
       "      <defs>\n",
       "       <path d=\"M 27.046875 69.921875 \n",
       "Q 40.625 69.921875 46.6875 58.734375 \n",
       "Q 51.375 50.09375 51.375 35.0625 \n",
       "Q 51.375 20.796875 47.125 11.46875 \n",
       "Q 40.96875 -1.90625 27 -1.90625 \n",
       "Q 14.40625 -1.90625 8.25 9.03125 \n",
       "Q 3.125 18.171875 3.125 33.546875 \n",
       "Q 3.125 45.453125 6.203125 54 \n",
       "Q 11.96875 69.921875 27.046875 69.921875 \n",
       "z\n",
       "M 26.953125 6.109375 \n",
       "Q 33.796875 6.109375 37.84375 12.15625 \n",
       "Q 41.890625 18.21875 41.890625 34.71875 \n",
       "Q 41.890625 46.625 38.953125 54.3125 \n",
       "Q 36.03125 62.015625 27.59375 62.015625 \n",
       "Q 19.828125 62.015625 16.234375 54.703125 \n",
       "Q 12.640625 47.40625 12.640625 33.203125 \n",
       "Q 12.640625 22.515625 14.9375 16.015625 \n",
       "Q 18.453125 6.109375 26.953125 6.109375 \n",
       "z\n",
       "\" id=\"Helvetica-30\"/>\n",
       "      </defs>\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(21.017031 121.010531)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-30\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_2\">\n",
       "     <g id=\"line2d_4\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"30.353594\" xlink:href=\"#m95115c8f30\" y=\"105.4497\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_4\">\n",
       "      <!-- 2 -->\n",
       "      <defs>\n",
       "       <path d=\"M 3.125 0 \n",
       "Q 3.609375 9.03125 6.859375 15.71875 \n",
       "Q 10.109375 22.40625 19.53125 27.875 \n",
       "L 28.90625 33.296875 \n",
       "Q 35.203125 36.96875 37.75 39.546875 \n",
       "Q 41.75 43.609375 41.75 48.828125 \n",
       "Q 41.75 54.9375 38.078125 58.515625 \n",
       "Q 34.421875 62.109375 28.328125 62.109375 \n",
       "Q 19.28125 62.109375 15.828125 55.28125 \n",
       "Q 13.96875 51.609375 13.765625 45.125 \n",
       "L 4.828125 45.125 \n",
       "Q 4.984375 54.25 8.203125 60.015625 \n",
       "Q 13.921875 70.171875 28.375 70.171875 \n",
       "Q 40.375 70.171875 45.921875 63.671875 \n",
       "Q 51.46875 57.171875 51.46875 49.21875 \n",
       "Q 51.46875 40.828125 45.5625 34.859375 \n",
       "Q 42.140625 31.390625 33.296875 26.46875 \n",
       "L 26.609375 22.75 \n",
       "Q 21.828125 20.125 19.09375 17.71875 \n",
       "Q 14.203125 13.484375 12.9375 8.296875 \n",
       "L 51.125 8.296875 \n",
       "L 51.125 0 \n",
       "L 3.125 0 \n",
       "z\n",
       "\" id=\"Helvetica-32\"/>\n",
       "      </defs>\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(21.017031 107.601731)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-32\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_3\">\n",
       "     <g id=\"line2d_5\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"30.353594\" xlink:href=\"#m95115c8f30\" y=\"92.0409\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_5\">\n",
       "      <!-- 4 -->\n",
       "      <defs>\n",
       "       <path d=\"M 33.0625 24.75 \n",
       "L 33.0625 56.453125 \n",
       "L 10.640625 24.75 \n",
       "z\n",
       "M 33.203125 0 \n",
       "L 33.203125 17.09375 \n",
       "L 2.546875 17.09375 \n",
       "L 2.546875 25.6875 \n",
       "L 34.578125 70.125 \n",
       "L 42 70.125 \n",
       "L 42 24.75 \n",
       "L 52.296875 24.75 \n",
       "L 52.296875 17.09375 \n",
       "L 42 17.09375 \n",
       "L 42 0 \n",
       "z\n",
       "\" id=\"Helvetica-34\"/>\n",
       "      </defs>\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(21.017031 94.192931)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-34\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_4\">\n",
       "     <g id=\"line2d_6\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"30.353594\" xlink:href=\"#m95115c8f30\" y=\"78.6321\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_6\">\n",
       "      <!-- 6 -->\n",
       "      <defs>\n",
       "       <path d=\"M 29.25 70.21875 \n",
       "Q 40.96875 70.21875 45.578125 64.140625 \n",
       "Q 50.203125 58.0625 50.203125 51.609375 \n",
       "L 41.5 51.609375 \n",
       "Q 40.71875 55.765625 39.015625 58.109375 \n",
       "Q 35.84375 62.5 29.390625 62.5 \n",
       "Q 22.015625 62.5 17.671875 55.6875 \n",
       "Q 13.328125 48.875 12.84375 36.1875 \n",
       "Q 15.875 40.625 20.453125 42.828125 \n",
       "Q 24.65625 44.78125 29.828125 44.78125 \n",
       "Q 38.625 44.78125 45.15625 39.15625 \n",
       "Q 51.703125 33.546875 51.703125 22.40625 \n",
       "Q 51.703125 12.890625 45.5 5.53125 \n",
       "Q 39.3125 -1.8125 27.828125 -1.8125 \n",
       "Q 18.015625 -1.8125 10.890625 5.640625 \n",
       "Q 3.765625 13.09375 3.765625 30.71875 \n",
       "Q 3.765625 43.75 6.9375 52.828125 \n",
       "Q 13.03125 70.21875 29.25 70.21875 \n",
       "z\n",
       "M 28.609375 6 \n",
       "Q 35.546875 6 38.984375 10.65625 \n",
       "Q 42.4375 15.328125 42.4375 21.6875 \n",
       "Q 42.4375 27.046875 39.359375 31.90625 \n",
       "Q 36.28125 36.765625 28.171875 36.765625 \n",
       "Q 22.515625 36.765625 18.234375 33 \n",
       "Q 13.96875 29.25 13.96875 21.6875 \n",
       "Q 13.96875 15.046875 17.84375 10.515625 \n",
       "Q 21.734375 6 28.609375 6 \n",
       "z\n",
       "\" id=\"Helvetica-36\"/>\n",
       "      </defs>\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(21.017031 80.784131)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-36\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_5\">\n",
       "     <g id=\"line2d_7\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"30.353594\" xlink:href=\"#m95115c8f30\" y=\"65.2233\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_7\">\n",
       "      <!-- 8 -->\n",
       "      <defs>\n",
       "       <path d=\"M 27.203125 40.625 \n",
       "Q 33.015625 40.625 36.28125 43.875 \n",
       "Q 39.546875 47.125 39.546875 51.609375 \n",
       "Q 39.546875 55.515625 36.421875 58.78125 \n",
       "Q 33.296875 62.0625 26.90625 62.0625 \n",
       "Q 20.5625 62.0625 17.71875 58.78125 \n",
       "Q 14.890625 55.515625 14.890625 51.125 \n",
       "Q 14.890625 46.1875 18.546875 43.40625 \n",
       "Q 22.21875 40.625 27.203125 40.625 \n",
       "z\n",
       "M 27.734375 6 \n",
       "Q 33.84375 6 37.859375 9.296875 \n",
       "Q 41.890625 12.59375 41.890625 19.140625 \n",
       "Q 41.890625 25.921875 37.734375 29.4375 \n",
       "Q 33.59375 32.953125 27.09375 32.953125 \n",
       "Q 20.796875 32.953125 16.8125 29.359375 \n",
       "Q 12.84375 25.78125 12.84375 19.4375 \n",
       "Q 12.84375 13.96875 16.484375 9.984375 \n",
       "Q 20.125 6 27.734375 6 \n",
       "z\n",
       "M 15.234375 37.25 \n",
       "Q 11.578125 38.8125 9.515625 40.921875 \n",
       "Q 5.671875 44.828125 5.671875 51.078125 \n",
       "Q 5.671875 58.890625 11.328125 64.5 \n",
       "Q 17 70.125 27.390625 70.125 \n",
       "Q 37.453125 70.125 43.15625 64.8125 \n",
       "Q 48.875 59.515625 48.875 52.4375 \n",
       "Q 48.875 45.90625 45.5625 41.84375 \n",
       "Q 43.703125 39.546875 39.796875 37.359375 \n",
       "Q 44.140625 35.359375 46.625 32.765625 \n",
       "Q 51.265625 27.875 51.265625 20.0625 \n",
       "Q 51.265625 10.84375 45.0625 4.421875 \n",
       "Q 38.875 -2 27.546875 -2 \n",
       "Q 17.328125 -2 10.265625 3.53125 \n",
       "Q 3.21875 9.078125 3.21875 19.625 \n",
       "Q 3.21875 25.828125 6.25 30.34375 \n",
       "Q 9.28125 34.859375 15.234375 37.25 \n",
       "z\n",
       "\" id=\"Helvetica-38\"/>\n",
       "      </defs>\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(21.017031 67.375331)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-38\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_6\">\n",
       "     <g id=\"line2d_8\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"30.353594\" xlink:href=\"#m95115c8f30\" y=\"51.8145\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_8\">\n",
       "      <!-- 10 -->\n",
       "      <defs>\n",
       "       <path d=\"M 9.578125 49.515625 \n",
       "L 9.578125 56.25 \n",
       "Q 19.09375 57.171875 22.84375 59.34375 \n",
       "Q 26.609375 61.53125 28.46875 69.625 \n",
       "L 35.40625 69.625 \n",
       "L 35.40625 0 \n",
       "L 26.03125 0 \n",
       "L 26.03125 49.515625 \n",
       "z\n",
       "\" id=\"Helvetica-31\"/>\n",
       "      </defs>\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(17.680469 53.966531)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-31\"/>\n",
       "       <use x=\"55.615234\" xlink:href=\"#Helvetica-30\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_7\">\n",
       "     <g id=\"line2d_9\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"30.353594\" xlink:href=\"#m95115c8f30\" y=\"38.4057\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_9\">\n",
       "      <!-- 12 -->\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(17.680469 40.557731)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-31\"/>\n",
       "       <use x=\"55.615234\" xlink:href=\"#Helvetica-32\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_8\">\n",
       "     <g id=\"line2d_10\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"30.353594\" xlink:href=\"#m95115c8f30\" y=\"24.9969\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_10\">\n",
       "      <!-- 14 -->\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(17.680469 27.148931)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-31\"/>\n",
       "       <use x=\"55.615234\" xlink:href=\"#Helvetica-34\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_11\">\n",
       "     <!-- % with endo cancer SNP within 5 kb -->\n",
       "     <defs>\n",
       "      <path d=\"M 68.171875 33.984375 \n",
       "Q 75.203125 33.984375 80.171875 29 \n",
       "Q 85.15625 24.03125 85.15625 17 \n",
       "Q 85.15625 9.96875 80.171875 4.984375 \n",
       "Q 75.203125 0 68.171875 0 \n",
       "Q 61.078125 0 56.09375 4.984375 \n",
       "Q 51.125 9.96875 51.125 17 \n",
       "Q 51.125 24.03125 56.09375 29 \n",
       "Q 61.078125 33.984375 68.171875 33.984375 \n",
       "z\n",
       "M 61.078125 69.625 \n",
       "L 66.5 69.625 \n",
       "L 27.390625 -1.859375 \n",
       "L 21.96875 -1.859375 \n",
       "z\n",
       "M 20.21875 40.71875 \n",
       "Q 24.421875 40.71875 27.375 43.671875 \n",
       "Q 30.328125 46.625 30.328125 50.78125 \n",
       "Q 30.328125 54.9375 27.375 57.890625 \n",
       "Q 24.421875 60.84375 20.21875 60.84375 \n",
       "Q 16.0625 60.84375 13.109375 57.90625 \n",
       "Q 10.15625 54.984375 10.15625 50.78125 \n",
       "Q 10.15625 46.625 13.109375 43.671875 \n",
       "Q 16.0625 40.71875 20.21875 40.71875 \n",
       "z\n",
       "M 20.21875 67.828125 \n",
       "Q 27.296875 67.828125 32.265625 62.8125 \n",
       "Q 37.25 57.8125 37.25 50.78125 \n",
       "Q 37.25 43.75 32.265625 38.765625 \n",
       "Q 27.296875 33.796875 20.21875 33.796875 \n",
       "Q 13.1875 33.796875 8.203125 38.765625 \n",
       "Q 3.21875 43.75 3.21875 50.78125 \n",
       "Q 3.21875 57.8125 8.203125 62.8125 \n",
       "Q 13.1875 67.828125 20.21875 67.828125 \n",
       "z\n",
       "M 68.171875 6.9375 \n",
       "Q 72.3125 6.9375 75.265625 9.890625 \n",
       "Q 78.21875 12.84375 78.21875 17 \n",
       "Q 78.21875 21.140625 75.265625 24.09375 \n",
       "Q 72.3125 27.046875 68.171875 27.046875 \n",
       "Q 63.96875 27.046875 61.015625 24.09375 \n",
       "Q 58.0625 21.140625 58.0625 17 \n",
       "Q 58.0625 12.84375 61.015625 9.890625 \n",
       "Q 63.96875 6.9375 68.171875 6.9375 \n",
       "z\n",
       "\" id=\"Helvetica-25\"/>\n",
       "      <path d=\"M 10.5 52.296875 \n",
       "L 20.5625 11.078125 \n",
       "L 30.765625 52.296875 \n",
       "L 40.625 52.296875 \n",
       "L 50.875 11.328125 \n",
       "L 61.578125 52.296875 \n",
       "L 70.359375 52.296875 \n",
       "L 55.171875 0 \n",
       "L 46.046875 0 \n",
       "L 35.40625 40.484375 \n",
       "L 25.09375 0 \n",
       "L 15.96875 0 \n",
       "L 0.875 52.296875 \n",
       "z\n",
       "\" id=\"Helvetica-77\"/>\n",
       "      <path d=\"M 12.015625 25.53125 \n",
       "Q 12.015625 17.140625 15.578125 11.46875 \n",
       "Q 19.140625 5.8125 27 5.8125 \n",
       "Q 33.109375 5.8125 37.03125 11.0625 \n",
       "Q 40.96875 16.3125 40.96875 26.125 \n",
       "Q 40.96875 36.03125 36.90625 40.796875 \n",
       "Q 32.859375 45.5625 26.90625 45.5625 \n",
       "Q 20.265625 45.5625 16.140625 40.484375 \n",
       "Q 12.015625 35.40625 12.015625 25.53125 \n",
       "z\n",
       "M 25.25 53.21875 \n",
       "Q 31.25 53.21875 35.296875 50.6875 \n",
       "Q 37.640625 49.21875 40.625 45.5625 \n",
       "L 40.625 71.96875 \n",
       "L 49.078125 71.96875 \n",
       "L 49.078125 0 \n",
       "L 41.15625 0 \n",
       "L 41.15625 7.28125 \n",
       "Q 38.09375 2.4375 33.890625 0.28125 \n",
       "Q 29.6875 -1.859375 24.265625 -1.859375 \n",
       "Q 15.53125 -1.859375 9.125 5.484375 \n",
       "Q 2.734375 12.84375 2.734375 25.046875 \n",
       "Q 2.734375 36.46875 8.5625 44.84375 \n",
       "Q 14.40625 53.21875 25.25 53.21875 \n",
       "z\n",
       "\" id=\"Helvetica-64\"/>\n",
       "      <path d=\"M 26.609375 53.8125 \n",
       "Q 35.453125 53.8125 40.984375 49.515625 \n",
       "Q 46.53125 45.21875 47.65625 34.71875 \n",
       "L 39.109375 34.71875 \n",
       "Q 38.328125 39.546875 35.546875 42.75 \n",
       "Q 32.765625 45.953125 26.609375 45.953125 \n",
       "Q 18.21875 45.953125 14.59375 37.75 \n",
       "Q 12.25 32.421875 12.25 24.609375 \n",
       "Q 12.25 16.75 15.5625 11.375 \n",
       "Q 18.890625 6 26.03125 6 \n",
       "Q 31.5 6 34.6875 9.34375 \n",
       "Q 37.890625 12.703125 39.109375 18.5 \n",
       "L 47.65625 18.5 \n",
       "Q 46.1875 8.109375 40.328125 3.296875 \n",
       "Q 34.46875 -1.515625 25.34375 -1.515625 \n",
       "Q 15.09375 -1.515625 8.984375 5.984375 \n",
       "Q 2.875 13.484375 2.875 24.703125 \n",
       "Q 2.875 38.484375 9.5625 46.140625 \n",
       "Q 16.265625 53.8125 26.609375 53.8125 \n",
       "z\n",
       "M 25.25 53.5625 \n",
       "z\n",
       "\" id=\"Helvetica-63\"/>\n",
       "      <path d=\"M 13.1875 13.921875 \n",
       "Q 13.1875 10.109375 15.96875 7.90625 \n",
       "Q 18.75 5.71875 22.5625 5.71875 \n",
       "Q 27.203125 5.71875 31.546875 7.859375 \n",
       "Q 38.875 11.421875 38.875 19.53125 \n",
       "L 38.875 26.609375 \n",
       "Q 37.25 25.59375 34.703125 24.90625 \n",
       "Q 32.171875 24.21875 29.734375 23.921875 \n",
       "L 24.421875 23.25 \n",
       "Q 19.625 22.609375 17.234375 21.234375 \n",
       "Q 13.1875 18.953125 13.1875 13.921875 \n",
       "z\n",
       "M 34.46875 31.6875 \n",
       "Q 37.5 32.078125 38.53125 34.234375 \n",
       "Q 39.109375 35.40625 39.109375 37.59375 \n",
       "Q 39.109375 42.09375 35.90625 44.109375 \n",
       "Q 32.71875 46.140625 26.765625 46.140625 \n",
       "Q 19.875 46.140625 17 42.4375 \n",
       "Q 15.375 40.375 14.890625 36.328125 \n",
       "L 6.6875 36.328125 \n",
       "Q 6.9375 46 12.96875 49.78125 \n",
       "Q 19 53.5625 26.953125 53.5625 \n",
       "Q 36.1875 53.5625 41.9375 50.046875 \n",
       "Q 47.65625 46.53125 47.65625 39.109375 \n",
       "L 47.65625 8.984375 \n",
       "Q 47.65625 7.625 48.21875 6.78125 \n",
       "Q 48.78125 5.953125 50.59375 5.953125 \n",
       "Q 51.171875 5.953125 51.90625 6.03125 \n",
       "Q 52.640625 6.109375 53.46875 6.25 \n",
       "L 53.46875 -0.25 \n",
       "Q 51.421875 -0.828125 50.34375 -0.96875 \n",
       "Q 49.265625 -1.125 47.40625 -1.125 \n",
       "Q 42.875 -1.125 40.828125 2.09375 \n",
       "Q 39.75 3.8125 39.3125 6.9375 \n",
       "Q 36.625 3.421875 31.59375 0.828125 \n",
       "Q 26.5625 -1.765625 20.515625 -1.765625 \n",
       "Q 13.234375 -1.765625 8.609375 2.65625 \n",
       "Q 4 7.078125 4 13.71875 \n",
       "Q 4 21 8.546875 25 \n",
       "Q 13.09375 29 20.453125 29.9375 \n",
       "z\n",
       "M 27.203125 53.5625 \n",
       "z\n",
       "\" id=\"Helvetica-61\"/>\n",
       "      <path d=\"M 13.96875 23.140625 \n",
       "Q 14.3125 17.046875 16.84375 13.234375 \n",
       "Q 21.6875 6.109375 33.890625 6.109375 \n",
       "Q 39.359375 6.109375 43.84375 7.671875 \n",
       "Q 52.546875 10.6875 52.546875 18.5 \n",
       "Q 52.546875 24.359375 48.875 26.859375 \n",
       "Q 45.171875 29.296875 37.25 31.109375 \n",
       "L 27.546875 33.296875 \n",
       "Q 18.015625 35.453125 14.0625 38.03125 \n",
       "Q 7.234375 42.53125 7.234375 51.46875 \n",
       "Q 7.234375 61.140625 13.921875 67.328125 \n",
       "Q 20.609375 73.53125 32.859375 73.53125 \n",
       "Q 44.140625 73.53125 52.015625 68.078125 \n",
       "Q 59.90625 62.640625 59.90625 50.6875 \n",
       "L 50.78125 50.6875 \n",
       "Q 50.046875 56.453125 47.65625 59.515625 \n",
       "Q 43.21875 65.140625 32.5625 65.140625 \n",
       "Q 23.96875 65.140625 20.203125 61.515625 \n",
       "Q 16.453125 57.90625 16.453125 53.125 \n",
       "Q 16.453125 47.859375 20.84375 45.40625 \n",
       "Q 23.734375 43.84375 33.890625 41.5 \n",
       "L 43.953125 39.203125 \n",
       "Q 51.21875 37.546875 55.171875 34.671875 \n",
       "Q 62.015625 29.640625 62.015625 20.0625 \n",
       "Q 62.015625 8.15625 53.34375 3.03125 \n",
       "Q 44.671875 -2.09375 33.203125 -2.09375 \n",
       "Q 19.828125 -2.09375 12.25 4.734375 \n",
       "Q 4.6875 11.53125 4.828125 23.140625 \n",
       "z\n",
       "M 33.59375 73.6875 \n",
       "z\n",
       "\" id=\"Helvetica-53\"/>\n",
       "      <path d=\"M 7.625 71.734375 \n",
       "L 19.09375 71.734375 \n",
       "L 55.328125 13.625 \n",
       "L 55.328125 71.734375 \n",
       "L 64.546875 71.734375 \n",
       "L 64.546875 0 \n",
       "L 53.65625 0 \n",
       "L 16.890625 58.0625 \n",
       "L 16.890625 0 \n",
       "L 7.625 0 \n",
       "z\n",
       "M 35.453125 71.734375 \n",
       "z\n",
       "\" id=\"Helvetica-4e\"/>\n",
       "      <path d=\"M 8.546875 71.734375 \n",
       "L 40.828125 71.734375 \n",
       "Q 50.390625 71.734375 56.25 66.328125 \n",
       "Q 62.109375 60.9375 62.109375 51.171875 \n",
       "Q 62.109375 42.78125 56.875 36.546875 \n",
       "Q 51.65625 30.328125 40.828125 30.328125 \n",
       "L 18.265625 30.328125 \n",
       "L 18.265625 0 \n",
       "L 8.546875 0 \n",
       "z\n",
       "M 52.296875 51.125 \n",
       "Q 52.296875 59.03125 46.4375 61.859375 \n",
       "Q 43.21875 63.375 37.59375 63.375 \n",
       "L 18.265625 63.375 \n",
       "L 18.265625 38.53125 \n",
       "L 37.59375 38.53125 \n",
       "Q 44.140625 38.53125 48.21875 41.3125 \n",
       "Q 52.296875 44.09375 52.296875 51.125 \n",
       "z\n",
       "\" id=\"Helvetica-50\"/>\n",
       "      <path d=\"M 12.359375 17.828125 \n",
       "Q 13.234375 10.296875 19.34375 7.421875 \n",
       "Q 22.46875 5.953125 26.5625 5.953125 \n",
       "Q 34.375 5.953125 38.125 10.9375 \n",
       "Q 41.890625 15.921875 41.890625 21.96875 \n",
       "Q 41.890625 29.296875 37.421875 33.296875 \n",
       "Q 32.953125 37.3125 26.703125 37.3125 \n",
       "Q 22.171875 37.3125 18.921875 35.546875 \n",
       "Q 15.671875 33.796875 13.375 30.671875 \n",
       "L 5.765625 31.109375 \n",
       "L 11.078125 68.75 \n",
       "L 47.40625 68.75 \n",
       "L 47.40625 60.25 \n",
       "L 17.671875 60.25 \n",
       "L 14.703125 40.828125 \n",
       "Q 17.140625 42.671875 19.34375 43.609375 \n",
       "Q 23.25 45.21875 28.375 45.21875 \n",
       "Q 37.984375 45.21875 44.671875 39.015625 \n",
       "Q 51.375 32.8125 51.375 23.296875 \n",
       "Q 51.375 13.375 45.234375 5.796875 \n",
       "Q 39.109375 -1.765625 25.6875 -1.765625 \n",
       "Q 17.140625 -1.765625 10.5625 3.046875 \n",
       "Q 4 7.859375 3.21875 17.828125 \n",
       "z\n",
       "\" id=\"Helvetica-35\"/>\n",
       "      <path d=\"M 6.25 71.734375 \n",
       "L 14.703125 71.734375 \n",
       "L 14.703125 30.078125 \n",
       "L 37.25 52.296875 \n",
       "L 48.484375 52.296875 \n",
       "L 28.46875 32.71875 \n",
       "L 49.609375 0 \n",
       "L 38.375 0 \n",
       "L 22.078125 26.375 \n",
       "L 14.703125 19.625 \n",
       "L 14.703125 0 \n",
       "L 6.25 0 \n",
       "z\n",
       "\" id=\"Helvetica-6b\"/>\n",
       "      <path d=\"M 5.765625 71.96875 \n",
       "L 14.3125 71.96875 \n",
       "L 14.3125 45.953125 \n",
       "Q 17.1875 49.703125 21.1875 51.671875 \n",
       "Q 25.203125 53.65625 29.890625 53.65625 \n",
       "Q 39.65625 53.65625 45.734375 46.9375 \n",
       "Q 51.8125 40.234375 51.8125 27.15625 \n",
       "Q 51.8125 14.75 45.796875 6.546875 \n",
       "Q 39.796875 -1.65625 29.15625 -1.65625 \n",
       "Q 23.1875 -1.65625 19.09375 1.21875 \n",
       "Q 16.65625 2.9375 13.875 6.6875 \n",
       "L 13.875 0 \n",
       "L 5.765625 0 \n",
       "z\n",
       "M 28.609375 6.109375 \n",
       "Q 35.75 6.109375 39.28125 11.765625 \n",
       "Q 42.828125 17.4375 42.828125 26.703125 \n",
       "Q 42.828125 34.96875 39.28125 40.375 \n",
       "Q 35.75 45.796875 28.859375 45.796875 \n",
       "Q 22.859375 45.796875 18.328125 41.359375 \n",
       "Q 13.8125 36.921875 13.8125 26.703125 \n",
       "Q 13.8125 19.34375 15.671875 14.75 \n",
       "Q 19.140625 6.109375 28.609375 6.109375 \n",
       "z\n",
       "\" id=\"Helvetica-62\"/>\n",
       "     </defs>\n",
       "     <g style=\"fill:#262626;\" transform=\"translate(12.221406 124.795344)rotate(-90)scale(0.07 -0.07)\">\n",
       "      <use xlink:href=\"#Helvetica-25\"/>\n",
       "      <use x=\"88.916016\" xlink:href=\"#Helvetica-20\"/>\n",
       "      <use x=\"116.699219\" xlink:href=\"#Helvetica-77\"/>\n",
       "      <use x=\"188.916016\" xlink:href=\"#Helvetica-69\"/>\n",
       "      <use x=\"211.132812\" xlink:href=\"#Helvetica-74\"/>\n",
       "      <use x=\"238.916016\" xlink:href=\"#Helvetica-68\"/>\n",
       "      <use x=\"294.53125\" xlink:href=\"#Helvetica-20\"/>\n",
       "      <use x=\"322.314453\" xlink:href=\"#Helvetica-65\"/>\n",
       "      <use x=\"377.929688\" xlink:href=\"#Helvetica-6e\"/>\n",
       "      <use x=\"433.544922\" xlink:href=\"#Helvetica-64\"/>\n",
       "      <use x=\"489.160156\" xlink:href=\"#Helvetica-6f\"/>\n",
       "      <use x=\"544.775391\" xlink:href=\"#Helvetica-20\"/>\n",
       "      <use x=\"572.558594\" xlink:href=\"#Helvetica-63\"/>\n",
       "      <use x=\"622.558594\" xlink:href=\"#Helvetica-61\"/>\n",
       "      <use x=\"678.173828\" xlink:href=\"#Helvetica-6e\"/>\n",
       "      <use x=\"733.789062\" xlink:href=\"#Helvetica-63\"/>\n",
       "      <use x=\"783.789062\" xlink:href=\"#Helvetica-65\"/>\n",
       "      <use x=\"839.404297\" xlink:href=\"#Helvetica-72\"/>\n",
       "      <use x=\"872.705078\" xlink:href=\"#Helvetica-20\"/>\n",
       "      <use x=\"900.488281\" xlink:href=\"#Helvetica-53\"/>\n",
       "      <use x=\"967.1875\" xlink:href=\"#Helvetica-4e\"/>\n",
       "      <use x=\"1039.404297\" xlink:href=\"#Helvetica-50\"/>\n",
       "      <use x=\"1106.103516\" xlink:href=\"#Helvetica-20\"/>\n",
       "      <use x=\"1133.886719\" xlink:href=\"#Helvetica-77\"/>\n",
       "      <use x=\"1206.103516\" xlink:href=\"#Helvetica-69\"/>\n",
       "      <use x=\"1228.320312\" xlink:href=\"#Helvetica-74\"/>\n",
       "      <use x=\"1256.103516\" xlink:href=\"#Helvetica-68\"/>\n",
       "      <use x=\"1311.71875\" xlink:href=\"#Helvetica-69\"/>\n",
       "      <use x=\"1333.935547\" xlink:href=\"#Helvetica-6e\"/>\n",
       "      <use x=\"1389.550781\" xlink:href=\"#Helvetica-20\"/>\n",
       "      <use x=\"1417.333984\" xlink:href=\"#Helvetica-35\"/>\n",
       "      <use x=\"1472.949219\" xlink:href=\"#Helvetica-20\"/>\n",
       "      <use x=\"1500.732422\" xlink:href=\"#Helvetica-6b\"/>\n",
       "      <use x=\"1550.732422\" xlink:href=\"#Helvetica-62\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"patch_3\">\n",
       "    <path clip-path=\"url(#p8862570797)\" d=\"M 32.371156 118.8585 \n",
       "L 50.30504 118.8585 \n",
       "L 50.30504 104.647485 \n",
       "L 32.371156 104.647485 \n",
       "z\n",
       "\" style=\"fill:#808080;stroke:#ffffff;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_4\">\n",
       "    <path clip-path=\"url(#p8862570797)\" d=\"M 54.788511 118.8585 \n",
       "L 72.722395 118.8585 \n",
       "L 72.722395 55.007071 \n",
       "L 54.788511 55.007071 \n",
       "z\n",
       "\" style=\"stroke:#ffffff;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_11\">\n",
       "    <path clip-path=\"url(#p8862570797)\" d=\"M 0 0 \n",
       "\" style=\"fill:none;stroke:#424242;stroke-linecap:round;stroke-width:2.16;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_12\">\n",
       "    <path clip-path=\"url(#p8862570797)\" d=\"M 0 0 \n",
       "\" style=\"fill:none;stroke:#424242;stroke-linecap:round;stroke-width:2.16;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_13\">\n",
       "    <path clip-path=\"url(#p8862570797)\" d=\"M 45.821569 48.302671 \n",
       "L 45.821569 48.302671 \n",
       "L 59.271982 48.302671 \n",
       "L 59.271982 48.302671 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:round;stroke-width:0.5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_5\">\n",
       "    <path d=\"M 30.353594 118.8585 \n",
       "L 30.353594 18.2925 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_6\">\n",
       "    <path d=\"M 74.739957 118.8585 \n",
       "L 74.739957 18.2925 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_7\">\n",
       "    <path d=\"M 30.353594 118.8585 \n",
       "L 74.739957 118.8585 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_8\">\n",
       "    <path d=\"M 30.353594 18.2925 \n",
       "L 74.739957 18.2925 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "   </g>\n",
       "   <g id=\"text_12\">\n",
       "    <!-- * -->\n",
       "    <defs>\n",
       "     <path d=\"M 21.96875 71.734375 \n",
       "L 21.96875 60.84375 \n",
       "L 32.625 64.546875 \n",
       "L 34.578125 59.125 \n",
       "L 23.921875 55.671875 \n",
       "L 30.515625 46.484375 \n",
       "L 25.6875 43.015625 \n",
       "L 19.28125 52.546875 \n",
       "L 12.84375 43.015625 \n",
       "L 7.859375 46.484375 \n",
       "L 14.546875 55.671875 \n",
       "L 3.8125 59.125 \n",
       "L 5.765625 64.546875 \n",
       "L 16.3125 60.75 \n",
       "L 16.3125 71.734375 \n",
       "z\n",
       "\" id=\"Helvetica-2a\"/>\n",
       "    </defs>\n",
       "    <g transform=\"translate(51.18451 46.843609)scale(0.07 -0.07)\">\n",
       "     <use xlink:href=\"#Helvetica-2a\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_13\">\n",
       "    <!-- lncRNA_good_csf -->\n",
       "    <defs>\n",
       "     <path d=\"M 6.6875 71.734375 \n",
       "L 15.484375 71.734375 \n",
       "L 15.484375 0 \n",
       "L 6.6875 0 \n",
       "z\n",
       "\" id=\"Helvetica-6c\"/>\n",
       "     <path d=\"M 40.96875 38.875 \n",
       "Q 47.796875 38.875 51.78125 41.609375 \n",
       "Q 55.765625 44.34375 55.765625 51.46875 \n",
       "Q 55.765625 59.125 50.203125 61.921875 \n",
       "Q 47.21875 63.375 42.234375 63.375 \n",
       "L 18.5 63.375 \n",
       "L 18.5 38.875 \n",
       "z\n",
       "M 8.796875 71.734375 \n",
       "L 42 71.734375 \n",
       "Q 50.203125 71.734375 55.515625 69.34375 \n",
       "Q 65.625 64.75 65.625 52.390625 \n",
       "Q 65.625 45.953125 62.953125 41.84375 \n",
       "Q 60.296875 37.75 55.515625 35.25 \n",
       "Q 59.71875 33.546875 61.84375 30.765625 \n",
       "Q 63.96875 27.984375 64.203125 21.734375 \n",
       "L 64.546875 12.109375 \n",
       "Q 64.703125 8.015625 65.234375 6 \n",
       "Q 66.109375 2.59375 68.359375 1.609375 \n",
       "L 68.359375 0 \n",
       "L 56.453125 0 \n",
       "Q 55.953125 0.921875 55.65625 2.390625 \n",
       "Q 55.375 3.859375 55.171875 8.0625 \n",
       "L 54.59375 20.015625 \n",
       "Q 54.25 27.046875 49.359375 29.4375 \n",
       "Q 46.578125 30.765625 40.625 30.765625 \n",
       "L 18.5 30.765625 \n",
       "L 18.5 0 \n",
       "L 8.796875 0 \n",
       "z\n",
       "\" id=\"Helvetica-52\"/>\n",
       "     <path d=\"M 44.4375 29.390625 \n",
       "L 33.546875 61.078125 \n",
       "L 21.96875 29.390625 \n",
       "z\n",
       "M 28.46875 71.734375 \n",
       "L 39.453125 71.734375 \n",
       "L 65.484375 0 \n",
       "L 54.828125 0 \n",
       "L 47.5625 21.484375 \n",
       "L 19.1875 21.484375 \n",
       "L 11.421875 0 \n",
       "L 1.46875 0 \n",
       "z\n",
       "M 33.5 71.734375 \n",
       "z\n",
       "\" id=\"Helvetica-41\"/>\n",
       "     <path d=\"M 0 -12.5 \n",
       "L 0 -7.5625 \n",
       "L 55.609375 -7.5625 \n",
       "L 55.609375 -12.5 \n",
       "z\n",
       "\" id=\"Helvetica-5f\"/>\n",
       "     <path d=\"M 8.640625 60.25 \n",
       "Q 8.84375 65.71875 10.546875 68.265625 \n",
       "Q 13.625 72.75 22.40625 72.75 \n",
       "Q 23.25 72.75 24.125 72.703125 \n",
       "Q 25 72.65625 26.125 72.5625 \n",
       "L 26.125 64.546875 \n",
       "Q 24.75 64.65625 24.140625 64.671875 \n",
       "Q 23.53125 64.703125 23 64.703125 \n",
       "Q 19 64.703125 18.21875 62.625 \n",
       "Q 17.4375 60.546875 17.4375 52.046875 \n",
       "L 26.125 52.046875 \n",
       "L 26.125 45.125 \n",
       "L 17.328125 45.125 \n",
       "L 17.328125 0 \n",
       "L 8.640625 0 \n",
       "L 8.640625 45.125 \n",
       "L 1.375 45.125 \n",
       "L 1.375 52.046875 \n",
       "L 8.640625 52.046875 \n",
       "L 8.640625 60.25 \n",
       "z\n",
       "\" id=\"Helvetica-66\"/>\n",
       "    </defs>\n",
       "    <g style=\"fill:#262626;\" transform=\"translate(24.532557 12.2925)scale(0.07 -0.07)\">\n",
       "     <use xlink:href=\"#Helvetica-6c\"/>\n",
       "     <use x=\"22.216797\" xlink:href=\"#Helvetica-6e\"/>\n",
       "     <use x=\"77.832031\" xlink:href=\"#Helvetica-63\"/>\n",
       "     <use x=\"127.832031\" xlink:href=\"#Helvetica-52\"/>\n",
       "     <use x=\"200.048828\" xlink:href=\"#Helvetica-4e\"/>\n",
       "     <use x=\"272.265625\" xlink:href=\"#Helvetica-41\"/>\n",
       "     <use x=\"338.964844\" xlink:href=\"#Helvetica-5f\"/>\n",
       "     <use x=\"394.580078\" xlink:href=\"#Helvetica-67\"/>\n",
       "     <use x=\"450.195312\" xlink:href=\"#Helvetica-6f\"/>\n",
       "     <use x=\"505.810547\" xlink:href=\"#Helvetica-6f\"/>\n",
       "     <use x=\"561.425781\" xlink:href=\"#Helvetica-64\"/>\n",
       "     <use x=\"617.041016\" xlink:href=\"#Helvetica-5f\"/>\n",
       "     <use x=\"672.65625\" xlink:href=\"#Helvetica-63\"/>\n",
       "     <use x=\"722.65625\" xlink:href=\"#Helvetica-73\"/>\n",
       "     <use x=\"772.65625\" xlink:href=\"#Helvetica-66\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "  </g>\n",
       "  <g id=\"axes_2\">\n",
       "   <g id=\"patch_9\">\n",
       "    <path d=\"M 83.61723 118.8585 \n",
       "L 128.003594 118.8585 \n",
       "L 128.003594 18.2925 \n",
       "L 83.61723 18.2925 \n",
       "z\n",
       "\" style=\"fill:#ffffff;\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_3\">\n",
       "    <g id=\"xtick_3\">\n",
       "     <g id=\"line2d_14\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"99.085205\" xlink:href=\"#m21aee0c6b8\" y=\"118.8585\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_14\">\n",
       "      <!-- no hit -->\n",
       "      <g style=\"fill:#262626;\" transform=\"translate(87.822983 138.279829)rotate(-45)scale(0.06 -0.06)\">\n",
       "       <use xlink:href=\"#Helvetica-6e\"/>\n",
       "       <use x=\"55.615234\" xlink:href=\"#Helvetica-6f\"/>\n",
       "       <use x=\"111.230469\" xlink:href=\"#Helvetica-20\"/>\n",
       "       <use x=\"139.013672\" xlink:href=\"#Helvetica-68\"/>\n",
       "       <use x=\"194.628906\" xlink:href=\"#Helvetica-69\"/>\n",
       "       <use x=\"216.845703\" xlink:href=\"#Helvetica-74\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_4\">\n",
       "    <g id=\"ytick_9\">\n",
       "     <g id=\"line2d_15\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"83.61723\" xlink:href=\"#m95115c8f30\" y=\"118.8585\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_10\">\n",
       "     <g id=\"line2d_16\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"83.61723\" xlink:href=\"#m95115c8f30\" y=\"105.4497\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_11\">\n",
       "     <g id=\"line2d_17\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"83.61723\" xlink:href=\"#m95115c8f30\" y=\"92.0409\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_12\">\n",
       "     <g id=\"line2d_18\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"83.61723\" xlink:href=\"#m95115c8f30\" y=\"78.6321\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_13\">\n",
       "     <g id=\"line2d_19\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"83.61723\" xlink:href=\"#m95115c8f30\" y=\"65.2233\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_14\">\n",
       "     <g id=\"line2d_20\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"83.61723\" xlink:href=\"#m95115c8f30\" y=\"51.8145\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_15\">\n",
       "     <g id=\"line2d_21\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"83.61723\" xlink:href=\"#m95115c8f30\" y=\"38.4057\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_16\">\n",
       "     <g id=\"line2d_22\">\n",
       "      <g>\n",
       "       <use style=\"fill:#262626;stroke:#262626;\" x=\"83.61723\" xlink:href=\"#m95115c8f30\" y=\"24.9969\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"patch_10\">\n",
       "    <path clip-path=\"url(#p7c35222344)\" d=\"M 85.634792 118.8585 \n",
       "L 112.535619 118.8585 \n",
       "L 112.535619 106.756695 \n",
       "L 85.634792 106.756695 \n",
       "z\n",
       "\" style=\"fill:#808080;stroke:#ffffff;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_11\">\n",
       "    <path clip-path=\"url(#p7c35222344)\" d=\"z\n",
       "\" style=\"stroke:#ffffff;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_23\">\n",
       "    <path clip-path=\"url(#p7c35222344)\" d=\"M 0 0 \n",
       "\" style=\"fill:none;stroke:#424242;stroke-linecap:round;stroke-width:2.16;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_24\">\n",
       "    <path clip-path=\"url(#p7c35222344)\" d=\"M 0 0 \n",
       "\" style=\"fill:none;stroke:#424242;stroke-linecap:round;stroke-width:2.16;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_25\">\n",
       "    <path clip-path=\"url(#p7c35222344)\" d=\"M 105.810412 63.097515 \n",
       "L 105.810412 63.097515 \n",
       "L 125.986032 63.097515 \n",
       "L 125.986032 63.097515 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:round;stroke-width:0.5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_12\">\n",
       "    <path d=\"M 83.61723 118.8585 \n",
       "L 83.61723 18.2925 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_13\">\n",
       "    <path d=\"M 128.003594 118.8585 \n",
       "L 128.003594 18.2925 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_14\">\n",
       "    <path d=\"M 83.61723 118.8585 \n",
       "L 128.003594 118.8585 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_15\">\n",
       "    <path d=\"M 83.61723 18.2925 \n",
       "L 128.003594 18.2925 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "   </g>\n",
       "   <g id=\"text_15\">\n",
       "    <!-- n.s. -->\n",
       "    <defs>\n",
       "     <path d=\"M 8.546875 10.640625 \n",
       "L 18.75 10.640625 \n",
       "L 18.75 0 \n",
       "L 8.546875 0 \n",
       "z\n",
       "\" id=\"Helvetica-2e\"/>\n",
       "    </defs>\n",
       "    <g transform=\"translate(110.257206 61.638452)scale(0.07 -0.07)\">\n",
       "     <use xlink:href=\"#Helvetica-6e\"/>\n",
       "     <use x=\"55.615234\" xlink:href=\"#Helvetica-2e\"/>\n",
       "     <use x=\"83.398438\" xlink:href=\"#Helvetica-73\"/>\n",
       "     <use x=\"133.398438\" xlink:href=\"#Helvetica-2e\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_16\">\n",
       "    <!-- protein_coding -->\n",
       "    <defs>\n",
       "     <path d=\"M 28.515625 5.90625 \n",
       "Q 34.671875 5.90625 38.75 11.0625 \n",
       "Q 42.828125 16.21875 42.828125 26.46875 \n",
       "Q 42.828125 32.71875 41.015625 37.203125 \n",
       "Q 37.59375 45.84375 28.515625 45.84375 \n",
       "Q 19.390625 45.84375 16.015625 36.71875 \n",
       "Q 14.203125 31.84375 14.203125 24.3125 \n",
       "Q 14.203125 18.265625 16.015625 14.015625 \n",
       "Q 19.4375 5.90625 28.515625 5.90625 \n",
       "z\n",
       "M 5.765625 52.046875 \n",
       "L 14.3125 52.046875 \n",
       "L 14.3125 45.125 \n",
       "Q 16.9375 48.6875 20.0625 50.640625 \n",
       "Q 24.515625 53.5625 30.515625 53.5625 \n",
       "Q 39.40625 53.5625 45.609375 46.75 \n",
       "Q 51.8125 39.9375 51.8125 27.296875 \n",
       "Q 51.8125 10.203125 42.875 2.875 \n",
       "Q 37.203125 -1.765625 29.6875 -1.765625 \n",
       "Q 23.78125 -1.765625 19.78125 0.828125 \n",
       "Q 17.4375 2.296875 14.546875 5.859375 \n",
       "L 14.546875 -20.84375 \n",
       "L 5.765625 -20.84375 \n",
       "z\n",
       "\" id=\"Helvetica-70\"/>\n",
       "    </defs>\n",
       "    <g style=\"fill:#262626;\" transform=\"translate(82.850412 12.2925)scale(0.07 -0.07)\">\n",
       "     <use xlink:href=\"#Helvetica-70\"/>\n",
       "     <use x=\"55.615234\" xlink:href=\"#Helvetica-72\"/>\n",
       "     <use x=\"88.916016\" xlink:href=\"#Helvetica-6f\"/>\n",
       "     <use x=\"144.53125\" xlink:href=\"#Helvetica-74\"/>\n",
       "     <use x=\"172.314453\" xlink:href=\"#Helvetica-65\"/>\n",
       "     <use x=\"227.929688\" xlink:href=\"#Helvetica-69\"/>\n",
       "     <use x=\"250.146484\" xlink:href=\"#Helvetica-6e\"/>\n",
       "     <use x=\"305.761719\" xlink:href=\"#Helvetica-5f\"/>\n",
       "     <use x=\"361.376953\" xlink:href=\"#Helvetica-63\"/>\n",
       "     <use x=\"411.376953\" xlink:href=\"#Helvetica-6f\"/>\n",
       "     <use x=\"466.992188\" xlink:href=\"#Helvetica-64\"/>\n",
       "     <use x=\"522.607422\" xlink:href=\"#Helvetica-69\"/>\n",
       "     <use x=\"544.824219\" xlink:href=\"#Helvetica-6e\"/>\n",
       "     <use x=\"600.439453\" xlink:href=\"#Helvetica-67\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "  </g>\n",
       " </g>\n",
       " <defs>\n",
       "  <clipPath id=\"p8862570797\">\n",
       "   <rect height=\"100.566\" width=\"44.386364\" x=\"30.353594\" y=\"18.2925\"/>\n",
       "  </clipPath>\n",
       "  <clipPath id=\"p7c35222344\">\n",
       "   <rect height=\"100.566\" width=\"44.386364\" x=\"83.61723\" y=\"18.2925\"/>\n",
       "  </clipPath>\n",
       " </defs>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2af65f490748>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "x = 5\n",
    "\n",
    "for t in [\"endo\"]:\n",
    "    fig, axarr = plt.subplots(figsize=(1.75, 1.85), nrows=1, ncols=2, sharey=True)\n",
    "\n",
    "    for i, csf in enumerate(order):\n",
    "        ax = axarr[i]\n",
    "        sub = data[data[\"csf\"] == csf]\n",
    "        sub = sub[sub[\"gb_closest_all_snp_dist\"] != -1]\n",
    "\n",
    "        tots = sub.groupby(\"is_hit\")[\"gene_id\"].agg(\"count\").reset_index()\n",
    "        within_x = sub.groupby([\"is_hit\", \"%s_within_%s_gb\" % (t, x)])[\"gene_id\"].agg(\"count\").reset_index()\n",
    "        within_x = within_x[within_x[\"%s_within_%s_gb\" % (t, x)] == \"yes\"]\n",
    "        perc = tots.merge(within_x, on=\"is_hit\", how=\"left\")\n",
    "        perc[\"perc\"] = (perc[\"gene_id_y\"]/perc[\"gene_id_x\"])*100\n",
    "\n",
    "        # fisher's exact\n",
    "        perc[\"no_enh\"] = perc[\"gene_id_x\"] - perc[\"gene_id_y\"]\n",
    "        table = perc[perc[\"is_hit\"].isin(hue_order)][[\"is_hit\", \"no_enh\", \"gene_id_y\"]].set_index(\"is_hit\")\n",
    "        #print(table)\n",
    "        table.fillna(0, inplace=True)\n",
    "        o, p = stats.fisher_exact(table, alternative='greater')\n",
    "        print(p)\n",
    "\n",
    "\n",
    "        sns.barplot(data=perc, x=\"is_hit\", y=\"perc\", ax=ax, order=hue_order, palette=pal)\n",
    "\n",
    "        ax.set_xticklabels(hue_order, rotation=45, ha=\"right\", va=\"top\")\n",
    "        ax.set_xlabel(\"\")\n",
    "        ax.set_ylabel(\"%% with %s cancer SNP within %s kb\" % (t, x))\n",
    "        ax.set_title(csf)\n",
    "        if i != 0:\n",
    "            ax.set_ylabel(\"\")\n",
    "        \n",
    "        if t == \"gwas\":\n",
    "            ax.set_ylim((0, 70))\n",
    "        elif t == \"cancer\":\n",
    "            ax.set_ylim((0, 25))\n",
    "        else:\n",
    "            ax.set_ylim((0, 15))\n",
    "            \n",
    "        # annotate pval\n",
    "        max_p = perc[\"perc\"].max()\n",
    "        annotate_pval(ax, 0.2, 0.8, max_p + 1, 0, max_p + 1, p, fontsize)\n",
    "\n",
    "        fig.savefig(\"Fig5C.pdf\", dpi=\"figure\", bbox_inches=\"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## write file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12946"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta.to_csv(\"../../../data/02__screen/02__enrichment_data/transcript_features.txt\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:py3]",
   "language": "python",
   "name": "conda-env-py3-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
